{"cells":[{"cell_type":"markdown","metadata":{"id":"CGCZUMbEIMmn"},"source":["# **ATAC-seq: Assay for Transposase-Accessible Chromatin**\n","## **What is ATAC-seq?**\n","\n","ATAC-seq is a fast and sensitive method used to study chromatin accessibility by identifying open chromatin regions across the genome.\n","\n","Instead of using antibodies like in ChIP-seq, ATAC-seq relies on a transposase enzyme (Tn5), which inserts sequencing adapters into open chromatin regions.\n","\n","**Applications of ATAC-seq**\n","\n","    Identifying Regulatory Elements → Finds promoters, enhancers, and open chromatin regions.\n","    Studying Chromatin Accessibility Changes → Detects differences between active and inactive genes.\n","    Single-Cell Epigenomics → Applied to scATAC-seq for cell-type-specific chromatin states.\n","    Cancer Research & Immunology → Helps identify epigenetic drivers of diseases.\n","\n","**ATAC-seq Workflow**\n","\n","    Cell Lysis & Nuclei Isolation\n","        Cells are lysed, and nuclei are extracted.\n","\n","    Tn5 Transposase Treatment\n","        Tn5 transposase inserts adapters into open chromatin regions.\n","        This step is called “tagmentation”.\n","\n","    Library Preparation & Sequencing\n","        Tagmented DNA fragments are PCR-amplified and sequenced.\n","\n","    Data Processing & Peak Calling\n","        Reads are aligned to the reference genome.\n","        Peak callers identify open chromatin regions.\n","\n","## **Analysis Plan**\n","\n","I will follow a structured pipeline to analyze the ATAC-seq data in Google Colab:\n","1. **Data Acquisition**\n","\n","    Download Corin-set-1 and Corin-set-2 FASTQ files from SRA using fasterq-dump (or another method if needed).\n","    Convert SRA format to paired-end FASTQ files.\n","\n","2. **Quality Control (QC)**\n","\n","    Assess read quality using FastQC.\n","    Identify any low-quality bases, adapter sequences, or sequencing artifacts.\n","\n","3. **Preprocessing & Trimming**\n","\n","    Trim low-quality bases and adapters using Trim Galore.\n","\n","4. **Read Alignment**\n","\n","    Align reads to the GRCh38 human reference genome using Bowtie2, an aligner optimized for short reads.\n","\n","5. **Post-Alignment Processing**\n","\n","    Convert SAM to BAM, sort, and remove duplicates using Samtools.\n","    Filter for properly mapped reads.\n","\n","6. **Peak Calling**\n","\n","    Use MACS2 to identify open chromatin regions (peaks).\n","    Compare peak profiles between Corin-treated and control samples.\n","\n","7. **Peak Annotation & Functional Analysis**\n","\n","    Annotate peaks to nearby genes using ChIPseeker.\n","    Perform motif analysis to identify transcription factor binding sites using HOMER.\n","    Identify differential chromatin accessibility using DiffBind.\n","\n","8. **Visualization**\n","\n","    Generate IGV browser tracks to visualize peaks.\n","    Create heatmaps and meta-plots using deepTools."]},{"cell_type":"markdown","metadata":{"id":"EJ7f1UfCMPSI"},"source":[]},{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":247214,"status":"ok","timestamp":1741549133247,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"dlCyTeNZDkAV","outputId":"a5db650f-a1ea-49de-cb38-c12c10234f3f"},"outputs":[{"output_type":"stream","name":"stdout","text":["\r0% [Working]\r            \rGet:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n","Get:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n","Hit:3 http://archive.ubuntu.com/ubuntu jammy InRelease\n","Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n","Get:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n","Hit:6 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n","Get:7 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n","Get:8 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [1,369 kB]\n","Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n","Hit:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n","Get:11 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,731 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n","Get:13 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [2,682 kB]\n","Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,533 kB]\n","Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [3,934 kB]\n","Get:16 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [3,774 kB]\n","Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [64.2 kB]\n","Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [2,988 kB]\n","Get:19 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [56.4 kB]\n","Get:20 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,668 kB]\n","Fetched 28.2 MB in 4s (6,821 kB/s)\n","Reading package lists... Done\n","W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","Package star is not available, but is referred to by another package.\n","This may mean that the package is missing, has been obsoleted, or\n","is only available from another source\n","\n","E: Package 'star' has no installation candidate\n","Collecting multiqc\n","  Downloading multiqc-1.27.1-py3-none-any.whl.metadata (46 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.1/46.1 kB\u001b[0m \u001b[31m1.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hCollecting scanpy\n","  Downloading scanpy-1.11.0-py3-none-any.whl.metadata (9.5 kB)\n","Collecting macs2\n","  Downloading MACS2-2.2.9.1.tar.gz (2.0 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.0/2.0 MB\u001b[0m \u001b[31m24.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n","  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n","  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n","Collecting pybedtools\n","  Downloading pybedtools-0.11.0.tar.gz (12.5 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m89.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","Collecting deeptools\n","  Downloading deepTools-3.5.6-py3-none-any.whl.metadata (4.0 kB)\n","Collecting biopython\n","  Downloading biopython-1.85-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\n","Collecting anndata\n","  Downloading anndata-0.11.3-py3-none-any.whl.metadata (8.2 kB)\n","Requirement already satisfied: matplotlib in /usr/local/lib/python3.11/dist-packages (3.10.0)\n","Requirement already satisfied: seaborn in /usr/local/lib/python3.11/dist-packages (0.13.2)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (1.26.4)\n","Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (2.2.2)\n","Requirement already satisfied: scipy in /usr/local/lib/python3.11/dist-packages (1.13.1)\n","Requirement already satisfied: scikit-learn in /usr/local/lib/python3.11/dist-packages (1.6.1)\n","Requirement already satisfied: umap-learn in /usr/local/lib/python3.11/dist-packages (0.5.7)\n","Collecting leidenalg\n","  Downloading leidenalg-0.10.2-cp38-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (10 kB)\n","Collecting cutadapt\n","  Downloading cutadapt-5.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.4 kB)\n","Requirement already satisfied: click in /usr/local/lib/python3.11/dist-packages (from multiqc) (8.1.8)\n","Requirement already satisfied: humanize in /usr/local/lib/python3.11/dist-packages (from multiqc) (4.11.0)\n","Requirement already satisfied: importlib_metadata in /usr/local/lib/python3.11/dist-packages (from multiqc) (8.6.1)\n","Requirement already satisfied: jinja2>=3.0.0 in /usr/local/lib/python3.11/dist-packages (from multiqc) (3.1.5)\n","Collecting kaleido==0.2.1 (from multiqc)\n","  Downloading kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl.metadata (15 kB)\n","Requirement already satisfied: markdown in /usr/local/lib/python3.11/dist-packages (from multiqc) (3.7)\n","Requirement already satisfied: packaging in /usr/local/lib/python3.11/dist-packages (from multiqc) (24.2)\n","Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from multiqc) (2.32.3)\n","Requirement already satisfied: Pillow>=10 in /usr/local/lib/python3.11/dist-packages (from multiqc) (11.1.0)\n","Requirement already satisfied: plotly>=5.18 in /usr/local/lib/python3.11/dist-packages (from multiqc) (5.24.1)\n","Requirement already satisfied: pyyaml>=4 in /usr/local/lib/python3.11/dist-packages (from multiqc) (6.0.2)\n","Requirement already satisfied: rich>=10 in /usr/local/lib/python3.11/dist-packages (from multiqc) (13.9.4)\n","Collecting rich-click (from multiqc)\n","  Downloading rich_click-1.8.7-py3-none-any.whl.metadata (7.9 kB)\n","Collecting coloredlogs (from multiqc)\n","  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)\n","Collecting spectra>=0.0.10 (from multiqc)\n","  Downloading spectra-0.1.0-py3-none-any.whl.metadata (927 bytes)\n","Requirement already satisfied: pydantic>=2.7.0 in /usr/local/lib/python3.11/dist-packages (from multiqc) (2.10.6)\n","Requirement already satisfied: typeguard in /usr/local/lib/python3.11/dist-packages (from multiqc) (4.4.2)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from multiqc) (4.67.1)\n","Collecting python-dotenv (from multiqc)\n","  Downloading python_dotenv-1.0.1-py3-none-any.whl.metadata (23 kB)\n","Requirement already satisfied: natsort in /usr/local/lib/python3.11/dist-packages (from multiqc) (8.4.0)\n","Collecting tiktoken (from multiqc)\n","  Downloading tiktoken-0.9.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.7 kB)\n","Requirement already satisfied: h5py>=3.7 in /usr/local/lib/python3.11/dist-packages (from scanpy) (3.12.1)\n","Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from scanpy) (1.4.2)\n","Collecting legacy-api-wrap>=1.4 (from scanpy)\n","  Downloading legacy_api_wrap-1.4.1-py3-none-any.whl.metadata (2.1 kB)\n","Requirement already satisfied: networkx>=2.7 in /usr/local/lib/python3.11/dist-packages (from scanpy) (3.4.2)\n","Requirement already satisfied: numba>=0.57 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.60.0)\n","Requirement already satisfied: patsy!=1.0.0 in /usr/local/lib/python3.11/dist-packages (from scanpy) (1.0.1)\n","Requirement already satisfied: pynndescent>=0.5 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.5.13)\n","Collecting scikit-learn\n","  Downloading scikit_learn-1.5.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\n","Collecting session-info2 (from scanpy)\n","  Downloading session_info2-0.1.2-py3-none-any.whl.metadata (2.5 kB)\n","Requirement already satisfied: statsmodels>=0.13 in /usr/local/lib/python3.11/dist-packages (from scanpy) (0.14.4)\n","Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (from scanpy) (4.12.2)\n","Collecting Cython~=0.29 (from macs2)\n","  Using cached Cython-0.29.37-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl.metadata (3.1 kB)\n","Collecting pysam (from pybedtools)\n","  Downloading pysam-0.23.0-cp311-cp311-manylinux_2_28_x86_64.whl.metadata (1.6 kB)\n","Collecting numpy\n","  Using cached numpy-2.2.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (62 kB)\n","Collecting numpydoc>=0.5 (from deeptools)\n","  Downloading numpydoc-1.8.0-py3-none-any.whl.metadata (4.3 kB)\n","Collecting pyBigWig>=0.2.1 (from deeptools)\n","  Downloading pyBigWig-0.3.24-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (19 kB)\n","Collecting py2bit>=0.2.0 (from deeptools)\n","  Downloading py2bit-0.3.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.0 kB)\n","Collecting deeptoolsintervals>=0.1.8 (from deeptools)\n","  Downloading deeptoolsintervals-0.1.9.tar.gz (41 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.1/41.1 kB\u001b[0m \u001b[31m2.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","Collecting array-api-compat!=1.5,>1.4 (from anndata)\n","  Downloading array_api_compat-1.11.1-py3-none-any.whl.metadata (1.8 kB)\n","Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib) (1.3.1)\n","Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib) (0.12.1)\n","Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib) (4.56.0)\n","Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib) (1.4.8)\n","Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib) (3.2.1)\n","Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.11/dist-packages (from matplotlib) (2.8.2)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas) (2025.1)\n","Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas) (2025.1)\n","Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn) (3.5.0)\n","Collecting igraph<0.12,>=0.10.0 (from leidenalg)\n","  Downloading igraph-0.11.8-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.8 kB)\n","Collecting dnaio>=1.2.3 (from cutadapt)\n","  Downloading dnaio-1.2.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.5 kB)\n","Collecting xopen>=1.6.0 (from cutadapt)\n","  Downloading xopen-2.0.2-py3-none-any.whl.metadata (15 kB)\n","Collecting texttable>=1.6.2 (from igraph<0.12,>=0.10.0->leidenalg)\n","  Downloading texttable-1.7.0-py2.py3-none-any.whl.metadata (9.8 kB)\n","Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2>=3.0.0->multiqc) (3.0.2)\n","Requirement already satisfied: llvmlite<0.44,>=0.43.0dev0 in /usr/local/lib/python3.11/dist-packages (from numba>=0.57->scanpy) (0.43.0)\n","Collecting numpy\n","  Downloading numpy-2.0.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m60.9/60.9 kB\u001b[0m \u001b[31m4.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: sphinx>=6 in /usr/local/lib/python3.11/dist-packages (from numpydoc>=0.5->deeptools) (8.1.3)\n","Requirement already satisfied: tabulate>=0.8.10 in /usr/local/lib/python3.11/dist-packages (from numpydoc>=0.5->deeptools) (0.9.0)\n","Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.11/dist-packages (from plotly>=5.18->multiqc) (9.0.0)\n","Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.7.0->multiqc) (0.7.0)\n","Requirement already satisfied: pydantic-core==2.27.2 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.7.0->multiqc) (2.27.2)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.7->matplotlib) (1.17.0)\n","Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.11/dist-packages (from rich>=10->multiqc) (3.0.0)\n","Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.11/dist-packages (from rich>=10->multiqc) (2.18.0)\n","Collecting colormath2>=3.0.0 (from spectra>=0.0.10->multiqc)\n","  Downloading colormath2-3.0.3-py3-none-any.whl.metadata (2.9 kB)\n","Collecting isal>=1.6.1 (from xopen>=1.6.0->cutadapt)\n","  Downloading isal-1.7.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)\n","Collecting zlib-ng>=0.4.1 (from xopen>=1.6.0->cutadapt)\n","  Downloading zlib_ng-0.5.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.8 kB)\n","Collecting humanfriendly>=9.1 (from coloredlogs->multiqc)\n","  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n","Requirement already satisfied: zipp>=3.20 in /usr/local/lib/python3.11/dist-packages (from importlib_metadata->multiqc) (3.21.0)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (3.4.1)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (3.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (2.3.0)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->multiqc) (2025.1.31)\n","Requirement already satisfied: regex>=2022.1.18 in /usr/local/lib/python3.11/dist-packages (from tiktoken->multiqc) (2024.11.6)\n","Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.11/dist-packages (from markdown-it-py>=2.2.0->rich>=10->multiqc) (0.1.2)\n","Requirement already satisfied: sphinxcontrib-applehelp>=1.0.7 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n","Requirement already satisfied: sphinxcontrib-devhelp>=1.0.6 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n","Requirement already satisfied: sphinxcontrib-htmlhelp>=2.0.6 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.1.0)\n","Requirement already satisfied: sphinxcontrib-jsmath>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (1.0.1)\n","Requirement already satisfied: sphinxcontrib-qthelp>=1.0.6 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n","Requirement already satisfied: sphinxcontrib-serializinghtml>=1.1.9 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n","Requirement already satisfied: docutils<0.22,>=0.20 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (0.21.2)\n","Requirement already satisfied: snowballstemmer>=2.2 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.2.0)\n","Requirement already satisfied: babel>=2.13 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.17.0)\n","Requirement already satisfied: alabaster>=0.7.14 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (1.0.0)\n","Requirement already satisfied: imagesize>=1.3 in /usr/local/lib/python3.11/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (1.4.1)\n","Downloading multiqc-1.27.1-py3-none-any.whl (4.7 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.7/4.7 MB\u001b[0m \u001b[31m92.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl (79.9 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m79.9/79.9 MB\u001b[0m \u001b[31m24.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading scanpy-1.11.0-py3-none-any.whl (2.1 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.1/2.1 MB\u001b[0m \u001b[31m67.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading deepTools-3.5.6-py3-none-any.whl (208 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m208.8/208.8 kB\u001b[0m \u001b[31m14.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading biopython-1.85-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.3 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m87.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading anndata-0.11.3-py3-none-any.whl (142 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m142.7/142.7 kB\u001b[0m \u001b[31m11.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading scikit_learn-1.5.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.3 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.3/13.3 MB\u001b[0m \u001b[31m99.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading leidenalg-0.10.2-cp38-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.0/2.0 MB\u001b[0m \u001b[31m2.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading cutadapt-5.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (286 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m286.1/286.1 kB\u001b[0m \u001b[31m20.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading array_api_compat-1.11.1-py3-none-any.whl (53 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m53.1/53.1 kB\u001b[0m \u001b[31m3.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hUsing cached Cython-0.29.37-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)\n","Downloading dnaio-1.2.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (108 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m108.9/108.9 kB\u001b[0m \u001b[31m7.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading igraph-0.11.8-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m83.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading legacy_api_wrap-1.4.1-py3-none-any.whl (10.0 kB)\n","Downloading numpy-2.0.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (19.5 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m19.5/19.5 MB\u001b[0m \u001b[31m76.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading numpydoc-1.8.0-py3-none-any.whl (64 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m64.0/64.0 kB\u001b[0m \u001b[31m4.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading py2bit-0.3.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (49 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m49.0/49.0 kB\u001b[0m \u001b[31m3.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading pyBigWig-0.3.24-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (185 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m185.8/185.8 kB\u001b[0m \u001b[31m13.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading pysam-0.23.0-cp311-cp311-manylinux_2_28_x86_64.whl (26.0 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m26.0/26.0 MB\u001b[0m \u001b[31m70.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading spectra-0.1.0-py3-none-any.whl (17 kB)\n","Downloading xopen-2.0.2-py3-none-any.whl (17 kB)\n","Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m3.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading python_dotenv-1.0.1-py3-none-any.whl (19 kB)\n","Downloading rich_click-1.8.7-py3-none-any.whl (35 kB)\n","Downloading session_info2-0.1.2-py3-none-any.whl (14 kB)\n","Downloading tiktoken-0.9.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.2 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m52.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading colormath2-3.0.3-py3-none-any.whl (40 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m40.7/40.7 kB\u001b[0m \u001b[31m2.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m6.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading isal-1.7.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (260 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m260.8/260.8 kB\u001b[0m \u001b[31m20.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading texttable-1.7.0-py2.py3-none-any.whl (10 kB)\n","Downloading zlib_ng-0.5.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (108 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m108.2/108.2 kB\u001b[0m \u001b[31m7.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hBuilding wheels for collected packages: macs2, pybedtools, deeptoolsintervals\n","  Building wheel for macs2 (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for macs2: filename=macs2-2.2.9.1-cp311-cp311-linux_x86_64.whl size=1828749 sha256=0db23d95ce9ed407b3d4982d6e62cbc556ccfa4b9bcd6da9a15074dbfe53618e\n","  Stored in directory: /root/.cache/pip/wheels/fa/c0/cd/e20c4ce750b1247dc1432a11eaca830948bab4e5326997636d\n","  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for pybedtools: filename=pybedtools-0.11.0-cp311-cp311-linux_x86_64.whl size=14028689 sha256=4accfefbad1a5213cbedd254203cf74b6eac3ed0604dacc0b60ac1691a9a3533\n","  Stored in directory: /root/.cache/pip/wheels/29/a9/64/b72ced6183351e9fb889c0fc86eda39fab286a4b8838443314\n","  Building wheel for deeptoolsintervals (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for deeptoolsintervals: filename=deeptoolsintervals-0.1.9-cp311-cp311-linux_x86_64.whl size=113439 sha256=afcfbeb016a18cd9d2e4523251eb81530f72fc2f707dd328187eee892812bae7\n","  Stored in directory: /root/.cache/pip/wheels/b5/09/8e/8589e68f9abaadf58a829c4415816154f5b1a43bd75faa13e7\n","Successfully built macs2 pybedtools deeptoolsintervals\n","Installing collected packages: texttable, py2bit, kaleido, deeptoolsintervals, zlib-ng, session-info2, python-dotenv, pysam, pyBigWig, numpy, legacy-api-wrap, isal, igraph, humanfriendly, Cython, array-api-compat, xopen, tiktoken, pybedtools, macs2, leidenalg, colormath2, coloredlogs, biopython, spectra, scikit-learn, rich-click, numpydoc, dnaio, anndata, multiqc, deeptools, cutadapt, scanpy\n","  Attempting uninstall: numpy\n","    Found existing installation: numpy 1.26.4\n","    Uninstalling numpy-1.26.4:\n","      Successfully uninstalled numpy-1.26.4\n","  Attempting uninstall: Cython\n","    Found existing installation: Cython 3.0.12\n","    Uninstalling Cython-3.0.12:\n","      Successfully uninstalled Cython-3.0.12\n","  Attempting uninstall: scikit-learn\n","    Found existing installation: scikit-learn 1.6.1\n","    Uninstalling scikit-learn-1.6.1:\n","      Successfully uninstalled scikit-learn-1.6.1\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","gensim 4.3.3 requires numpy<2.0,>=1.18.5, but you have numpy 2.0.2 which is incompatible.\n","langchain 0.3.19 requires numpy<2,>=1.26.4; python_version < \"3.12\", but you have numpy 2.0.2 which is incompatible.\n","thinc 8.2.5 requires numpy<2.0.0,>=1.19.0; python_version >= \"3.9\", but you have numpy 2.0.2 which is incompatible.\n","pytensor 2.27.1 requires numpy<2,>=1.17.0, but you have numpy 2.0.2 which is incompatible.\u001b[0m\u001b[31m\n","\u001b[0mSuccessfully installed Cython-0.29.37 anndata-0.11.3 array-api-compat-1.11.1 biopython-1.85 coloredlogs-15.0.1 colormath2-3.0.3 cutadapt-5.0 deeptools-3.5.6 deeptoolsintervals-0.1.9 dnaio-1.2.3 humanfriendly-10.0 igraph-0.11.8 isal-1.7.2 kaleido-0.2.1 legacy-api-wrap-1.4.1 leidenalg-0.10.2 macs2-2.2.9.1 multiqc-1.27.1 numpy-2.0.2 numpydoc-1.8.0 py2bit-0.3.3 pyBigWig-0.3.24 pybedtools-0.11.0 pysam-0.23.0 python-dotenv-1.0.1 rich-click-1.8.7 scanpy-1.11.0 scikit-learn-1.5.2 session-info2-0.1.2 spectra-0.1.0 texttable-1.7.0 tiktoken-0.9.0 xopen-2.0.2 zlib-ng-0.5.1\n","--2025-03-09 19:38:09--  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n","Resolving ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)... 130.14.250.11, 130.14.250.10, 2607:f220:41e:250::10, ...\n","Connecting to ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)|130.14.250.11|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 92236991 (88M) [application/x-gzip]\n","Saving to: ‘sratoolkit.current-ubuntu64.tar.gz’\n","\n","sratoolkit.current- 100%[===================>]  87.96M  22.2MB/s    in 5.3s    \n","\n","2025-03-09 19:38:16 (16.6 MB/s) - ‘sratoolkit.current-ubuntu64.tar.gz’ saved [92236991/92236991]\n","\n","sratoolkit.3.2.0-ubuntu64/\n","sratoolkit.3.2.0-ubuntu64/README.md\n","sratoolkit.3.2.0-ubuntu64/README-vdb-config\n","sratoolkit.3.2.0-ubuntu64/schema/\n","sratoolkit.3.2.0-ubuntu64/schema/vdb/\n","sratoolkit.3.2.0-ubuntu64/schema/vdb/vdb.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/vdb/built-in.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/insdc/\n","sratoolkit.3.2.0-ubuntu64/schema/insdc/insdc.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/insdc/sra.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/insdc/seq.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/\n","sratoolkit.3.2.0-ubuntu64/schema/sra/abi.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/pevents.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/generic-fastq.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/454.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/pacbio.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/illumina.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/helicos.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/nanopore.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/sra/ion-torrent.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/clip.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/seq-graph.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/stats.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/ncbi.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/wgs-contig.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/sra.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/spotname.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/pnbrdb.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/varloc.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/seq.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/ncbi/trace.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/align/\n","sratoolkit.3.2.0-ubuntu64/schema/align/refseq.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/align/pileup-stats.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/align/mate-cache.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/align/seq.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/align/align.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/align/qstat.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/csra2/\n","sratoolkit.3.2.0-ubuntu64/schema/csra2/stats.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/csra2/reference.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/csra2/read.vschema\n","sratoolkit.3.2.0-ubuntu64/schema/csra2/csra2.vschema\n","sratoolkit.3.2.0-ubuntu64/bin/\n","sratoolkit.3.2.0-ubuntu64/bin/sra-sort-cg.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/prefetch.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-dump-orig.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/kdbmeta.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/bam-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-search\n","sratoolkit.3.2.0-ubuntu64/bin/cache-mgr.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-validate.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/align-info.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/prefetch.3\n","sratoolkit.3.2.0-ubuntu64/bin/sam-dump.3\n","sratoolkit.3.2.0-ubuntu64/bin/sff-dump.3\n","sratoolkit.3.2.0-ubuntu64/bin/sff-dump\n","sratoolkit.3.2.0-ubuntu64/bin/fastq-dump\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-unlock.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/fastq-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/sam-dump.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/test-sra.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/sra-sort.3\n","sratoolkit.3.2.0-ubuntu64/bin/sratools\n","sratoolkit.3.2.0-ubuntu64/bin/cache-mgr.3\n","sratoolkit.3.2.0-ubuntu64/bin/cache-mgr\n","sratoolkit.3.2.0-ubuntu64/bin/srapath\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-dump.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/srapath.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/abi-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/kar.3\n","sratoolkit.3.2.0-ubuntu64/bin/srf-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/illumina-dump\n","sratoolkit.3.2.0-ubuntu64/bin/srapath.3\n","sratoolkit.3.2.0-ubuntu64/bin/sam-dump\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-encrypt\n","sratoolkit.3.2.0-ubuntu64/bin/rcexplain.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/rcexplain\n","sratoolkit.3.2.0-ubuntu64/bin/sff-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-decrypt\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-lock.3\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-lock\n","sratoolkit.3.2.0-ubuntu64/bin/ncbi/\n","sratoolkit.3.2.0-ubuntu64/bin/ncbi/default.kfg\n","sratoolkit.3.2.0-ubuntu64/bin/ncbi/certs.kfg\n","sratoolkit.3.2.0-ubuntu64/bin/ncbi/vdb-copy.kfg\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-validate\n","sratoolkit.3.2.0-ubuntu64/bin/rcexplain.3\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-copy.3\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-config\n","sratoolkit.3.2.0-ubuntu64/bin/fastq-load\n","sratoolkit.3.2.0-ubuntu64/bin/sff-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/pacbio-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/illumina-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-dump.3\n","sratoolkit.3.2.0-ubuntu64/bin/test-sra.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-sort.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/helicos-load\n","sratoolkit.3.2.0-ubuntu64/bin/pacbio-load\n","sratoolkit.3.2.0-ubuntu64/bin/helicos-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/abi-dump\n","sratoolkit.3.2.0-ubuntu64/bin/test-sra\n","sratoolkit.3.2.0-ubuntu64/bin/fastq-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-validate.3\n","sratoolkit.3.2.0-ubuntu64/bin/abi-dump.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/srapath-orig.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/illumina-dump.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-pileup.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/prefetch-orig.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-encrypt.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/bam-load\n","sratoolkit.3.2.0-ubuntu64/bin/sratools.3\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-decrypt.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/kdbmeta.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-sort-cg\n","sratoolkit.3.2.0-ubuntu64/bin/fasterq-dump.3\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-copy.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/latf-load\n","sratoolkit.3.2.0-ubuntu64/bin/abi-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/bam-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/fasterq-dump-orig.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/srf-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-unlock\n","sratoolkit.3.2.0-ubuntu64/bin/illumina-load\n","sratoolkit.3.2.0-ubuntu64/bin/srf-load\n","sratoolkit.3.2.0-ubuntu64/bin/helicos-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-encrypt.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-stat.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-pileup.3\n","sratoolkit.3.2.0-ubuntu64/bin/illumina-dump.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/align-info\n","sratoolkit.3.2.0-ubuntu64/bin/fasterq-dump.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/sra-search.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-stat.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/sra-search.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/illumina-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/abi-load\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-decrypt.3\n","sratoolkit.3.2.0-ubuntu64/bin/latf-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/fastq-dump.3\n","sratoolkit.3.2.0-ubuntu64/bin/fastq-dump-orig.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/kar\n","sratoolkit.3.2.0-ubuntu64/bin/sra-pileup-orig.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/cg-load\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-copy\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-config.3\n","sratoolkit.3.2.0-ubuntu64/bin/cg-load.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-config.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/sra-sort-cg.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-stat\n","sratoolkit.3.2.0-ubuntu64/bin/fastq-dump.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-lock.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/abi-dump.3\n","sratoolkit.3.2.0-ubuntu64/bin/align-info.3\n","sratoolkit.3.2.0-ubuntu64/bin/sra-sort\n","sratoolkit.3.2.0-ubuntu64/bin/kar.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/kdbmeta\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-dump\n","sratoolkit.3.2.0-ubuntu64/bin/sra-pileup\n","sratoolkit.3.2.0-ubuntu64/bin/sff-dump.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/latf-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/cg-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/prefetch\n","sratoolkit.3.2.0-ubuntu64/bin/sam-dump-orig.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/sratools.3.2.0\n","sratoolkit.3.2.0-ubuntu64/bin/fasterq-dump\n","sratoolkit.3.2.0-ubuntu64/bin/pacbio-load.3\n","sratoolkit.3.2.0-ubuntu64/bin/vdb-unlock.3\n","sratoolkit.3.2.0-ubuntu64/bin/sff-load\n","sratoolkit.3.2.0-ubuntu64/example/\n","sratoolkit.3.2.0-ubuntu64/example/perl/\n","sratoolkit.3.2.0-ubuntu64/example/perl/base-stats.pl\n","sratoolkit.3.2.0-ubuntu64/example/perl/mismatch-stats.pl\n","sratoolkit.3.2.0-ubuntu64/example/perl/gene-lookup.pl\n","sratoolkit.3.2.0-ubuntu64/example/perl/simplefastq.pl\n","sratoolkit.3.2.0-ubuntu64/example/perl/splitfastq.pl\n","sratoolkit.3.2.0-ubuntu64/example/perl/dump-reference.pl\n","sratoolkit.3.2.0-ubuntu64/example/perl/quality-stats.pl\n","sratoolkit.3.2.0-ubuntu64/CHANGES\n","sratoolkit.3.2.0-ubuntu64/README-blastn\n","Requirement already satisfied: cutadapt in /usr/local/lib/python3.11/dist-packages (5.0)\n","Requirement already satisfied: dnaio>=1.2.3 in /usr/local/lib/python3.11/dist-packages (from cutadapt) (1.2.3)\n","Requirement already satisfied: xopen>=1.6.0 in /usr/local/lib/python3.11/dist-packages (from cutadapt) (2.0.2)\n","Requirement already satisfied: isal>=1.6.1 in /usr/local/lib/python3.11/dist-packages (from xopen>=1.6.0->cutadapt) (1.7.2)\n","Requirement already satisfied: zlib-ng>=0.4.1 in /usr/local/lib/python3.11/dist-packages (from xopen>=1.6.0->cutadapt) (0.5.1)\n","TrimGalore-0.6.10/\n","TrimGalore-0.6.10/.travis.yml\n","TrimGalore-0.6.10/CHANGELOG.md\n","TrimGalore-0.6.10/Docs/\n","TrimGalore-0.6.10/Docs/Images/\n","TrimGalore-0.6.10/Docs/Images/Thumbs.db\n","TrimGalore-0.6.10/Docs/Images/adapters.png\n","TrimGalore-0.6.10/Docs/Images/adapters_fixed.png\n","TrimGalore-0.6.10/Docs/Images/fixed_quals.png\n","TrimGalore-0.6.10/Docs/Images/fixed_quals_per_sequence.png\n","TrimGalore-0.6.10/Docs/Images/logo.png\n","TrimGalore-0.6.10/Docs/Images/pigz_bench.png\n","TrimGalore-0.6.10/Docs/Images/poor_q_per_sequence.png\n","TrimGalore-0.6.10/Docs/Images/poor_quals.png\n","TrimGalore-0.6.10/Docs/RRBS_Guide.pdf\n","TrimGalore-0.6.10/Docs/Trim_Galore_User_Guide.md\n","TrimGalore-0.6.10/LICENSE\n","TrimGalore-0.6.10/README.md\n","TrimGalore-0.6.10/test_files/\n","TrimGalore-0.6.10/test_files/4_seqs_with_Ns.fastq.gz\n","TrimGalore-0.6.10/test_files/clock_10K_R1.fastq.gz\n","TrimGalore-0.6.10/test_files/clock_10K_R2.fastq.gz\n","TrimGalore-0.6.10/test_files/colorspace_file.fastq\n","TrimGalore-0.6.10/test_files/empty_file.fastq\n","TrimGalore-0.6.10/test_files/illumina_100K.fastq.gz\n","TrimGalore-0.6.10/test_files/illumina_100K.fastq.gz_trimming_report.txt\n","TrimGalore-0.6.10/test_files/illumina_10K.fastq.gz\n","TrimGalore-0.6.10/test_files/illumina_10K.fastq.gz_trimming_report.txt\n","TrimGalore-0.6.10/test_files/illumina_10K_trimmed.fq.gz\n","TrimGalore-0.6.10/test_files/nextera_100K.fastq.gz\n","TrimGalore-0.6.10/test_files/nextera_100K.fastq.gz_trimming_report.txt\n","TrimGalore-0.6.10/test_files/smallRNA_100K.fastq.gz\n","TrimGalore-0.6.10/test_files/smallRNA_100K.fastq.gz_trimming_report.txt\n","TrimGalore-0.6.10/test_files/smallRNA_100K_R1.fastq.gz_trimming_report.txt\n","TrimGalore-0.6.10/test_files/smallRNA_100K_R2.fastq.gz\n","TrimGalore-0.6.10/test_files/smallRNA_100K_R2.fastq.gz_trimming_report.txt\n","TrimGalore-0.6.10/test_files/truncated.fq.gz\n","TrimGalore-0.6.10/trim_galore\n","--2025-03-09 19:38:24--  http://prinseq.sourceforge.net/distributions/prinseq-lite-0.20.4.tar.gz\n","Resolving prinseq.sourceforge.net (prinseq.sourceforge.net)... 104.18.13.149, 104.18.12.149, 2606:4700::6812:d95, ...\n","Connecting to prinseq.sourceforge.net (prinseq.sourceforge.net)|104.18.13.149|:80... connected.\n","HTTP request sent, awaiting response... 301 Moved Permanently\n","Location: https://prinseq.sourceforge.net/distributions/prinseq-lite-0.20.4.tar.gz [following]\n","--2025-03-09 19:38:24--  https://prinseq.sourceforge.net/distributions/prinseq-lite-0.20.4.tar.gz\n","Connecting to prinseq.sourceforge.net (prinseq.sourceforge.net)|104.18.13.149|:443... connected.\n","HTTP request sent, awaiting response... 404 Not Found\n","2025-03-09 19:38:24 ERROR 404: Not Found.\n","\n","tar (child): prinseq-lite-0.20.4.tar.gz: Cannot open: No such file or directory\n","tar (child): Error is not recoverable: exiting now\n","tar: Child returned status 2\n","tar: Error is not recoverable: exiting now\n","mv: cannot stat 'prinseq-lite-0.20.4': No such file or directory\n","Requirement already satisfied: rpy2 in /usr/local/lib/python3.11/dist-packages (3.5.17)\n","Requirement already satisfied: cffi>=1.15.1 in /usr/local/lib/python3.11/dist-packages (from rpy2) (1.17.1)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from rpy2) (3.1.5)\n","Requirement already satisfied: tzlocal in /usr/local/lib/python3.11/dist-packages (from rpy2) (5.3)\n","Requirement already satisfied: pycparser in /usr/local/lib/python3.11/dist-packages (from cffi>=1.15.1->rpy2) (2.22)\n","Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->rpy2) (3.0.2)\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","r-base is already the newest version (4.4.3-1.2204.0).\n","0 upgraded, 0 newly installed, 0 to remove and 33 not upgraded.\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","The following additional packages will be installed:\n","  fonts-dejavu-core fonts-dejavu-extra libatk-wrapper-java libatk-wrapper-java-jni libfontenc1\n","  libice-dev libsm-dev libxkbfile1 libxt-dev libxtst6 libxxf86dga1 openjdk-17-jdk-headless\n","  openjdk-17-jre openjdk-17-jre-headless x11-utils\n","Suggested packages:\n","  libice-doc libsm-doc libxt-doc openjdk-17-demo openjdk-17-source visualvm libnss-mdns\n","  fonts-ipafont-gothic fonts-ipafont-mincho fonts-wqy-microhei | fonts-wqy-zenhei fonts-indic\n","  mesa-utils\n","The following NEW packages will be installed:\n","  fonts-dejavu-core fonts-dejavu-extra libatk-wrapper-java libatk-wrapper-java-jni libfontenc1\n","  libice-dev libsm-dev libxkbfile1 libxt-dev libxtst6 libxxf86dga1 openjdk-17-jdk\n","  openjdk-17-jdk-headless openjdk-17-jre openjdk-17-jre-headless x11-utils\n","0 upgraded, 16 newly installed, 0 to remove and 33 not upgraded.\n","Need to get 125 MB of archives.\n","After this operation, 287 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n","Get:3 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfontenc1 amd64 1:1.1.4-1build3 [14.7 kB]\n","Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbfile1 amd64 1:1.1.0-1build3 [71.8 kB]\n","Get:5 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n","Get:6 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n","Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n","Get:8 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n","Get:9 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n","Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 libice-dev amd64 2:1.0.10-1build2 [51.4 kB]\n","Get:11 http://archive.ubuntu.com/ubuntu jammy/main amd64 libsm-dev amd64 2:1.2.3-1build2 [18.1 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxt-dev amd64 1:1.2.1-1 [396 kB]\n","Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jre-headless amd64 17.0.14+7-1~22.04.1 [48.3 MB]\n","Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jre amd64 17.0.14+7-1~22.04.1 [232 kB]\n","Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jdk-headless amd64 17.0.14+7-1~22.04.1 [71.3 MB]\n","Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jdk amd64 17.0.14+7-1~22.04.1 [1,521 kB]\n","Fetched 125 MB in 8s (14.9 MB/s)\n","Selecting previously unselected package fonts-dejavu-core.\n","(Reading database ... 124947 files and directories currently installed.)\n","Preparing to unpack .../00-fonts-dejavu-core_2.37-2build1_all.deb ...\n","Unpacking fonts-dejavu-core (2.37-2build1) ...\n","Selecting previously unselected package fonts-dejavu-extra.\n","Preparing to unpack .../01-fonts-dejavu-extra_2.37-2build1_all.deb ...\n","Unpacking fonts-dejavu-extra (2.37-2build1) ...\n","Selecting previously unselected package libfontenc1:amd64.\n","Preparing to unpack .../02-libfontenc1_1%3a1.1.4-1build3_amd64.deb ...\n","Unpacking libfontenc1:amd64 (1:1.1.4-1build3) ...\n","Selecting previously unselected package libxkbfile1:amd64.\n","Preparing to unpack .../03-libxkbfile1_1%3a1.1.0-1build3_amd64.deb ...\n","Unpacking libxkbfile1:amd64 (1:1.1.0-1build3) ...\n","Selecting previously unselected package libxtst6:amd64.\n","Preparing to unpack .../04-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n","Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n","Selecting previously unselected package libxxf86dga1:amd64.\n","Preparing to unpack .../05-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n","Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n","Selecting previously unselected package x11-utils.\n","Preparing to unpack .../06-x11-utils_7.7+5build2_amd64.deb ...\n","Unpacking x11-utils (7.7+5build2) ...\n","Selecting previously unselected package libatk-wrapper-java.\n","Preparing to unpack .../07-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n","Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n","Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n","Preparing to unpack .../08-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n","Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n","Selecting previously unselected package libice-dev:amd64.\n","Preparing to unpack .../09-libice-dev_2%3a1.0.10-1build2_amd64.deb ...\n","Unpacking libice-dev:amd64 (2:1.0.10-1build2) ...\n","Selecting previously unselected package libsm-dev:amd64.\n","Preparing to unpack .../10-libsm-dev_2%3a1.2.3-1build2_amd64.deb ...\n","Unpacking libsm-dev:amd64 (2:1.2.3-1build2) ...\n","Selecting previously unselected package libxt-dev:amd64.\n","Preparing to unpack .../11-libxt-dev_1%3a1.2.1-1_amd64.deb ...\n","Unpacking libxt-dev:amd64 (1:1.2.1-1) ...\n","Selecting previously unselected package openjdk-17-jre-headless:amd64.\n","Preparing to unpack .../12-openjdk-17-jre-headless_17.0.14+7-1~22.04.1_amd64.deb ...\n","Unpacking openjdk-17-jre-headless:amd64 (17.0.14+7-1~22.04.1) ...\n","Selecting previously unselected package openjdk-17-jre:amd64.\n","Preparing to unpack .../13-openjdk-17-jre_17.0.14+7-1~22.04.1_amd64.deb ...\n","Unpacking openjdk-17-jre:amd64 (17.0.14+7-1~22.04.1) ...\n","Selecting previously unselected package openjdk-17-jdk-headless:amd64.\n","Preparing to unpack .../14-openjdk-17-jdk-headless_17.0.14+7-1~22.04.1_amd64.deb ...\n","Unpacking openjdk-17-jdk-headless:amd64 (17.0.14+7-1~22.04.1) ...\n","Selecting previously unselected package openjdk-17-jdk:amd64.\n","Preparing to unpack .../15-openjdk-17-jdk_17.0.14+7-1~22.04.1_amd64.deb ...\n","Unpacking openjdk-17-jdk:amd64 (17.0.14+7-1~22.04.1) ...\n","Setting up libice-dev:amd64 (2:1.0.10-1build2) ...\n","Setting up libsm-dev:amd64 (2:1.2.3-1build2) ...\n","Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n","Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n","Setting up libfontenc1:amd64 (1:1.1.4-1build3) ...\n","Setting up libxt-dev:amd64 (1:1.2.1-1) ...\n","Setting up fonts-dejavu-core (2.37-2build1) ...\n","Setting up openjdk-17-jre-headless:amd64 (17.0.14+7-1~22.04.1) ...\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/java to provide /usr/bin/java (java) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jpackage to provide /usr/bin/jpackage (jpackage) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/keytool to provide /usr/bin/keytool (keytool) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/rmiregistry to provide /usr/bin/rmiregistry (rmiregistry) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/lib/jexec to provide /usr/bin/jexec (jexec) in auto mode\n","Setting up fonts-dejavu-extra (2.37-2build1) ...\n","Setting up libxkbfile1:amd64 (1:1.1.0-1build3) ...\n","Setting up openjdk-17-jre:amd64 (17.0.14+7-1~22.04.1) ...\n","Setting up openjdk-17-jdk-headless:amd64 (17.0.14+7-1~22.04.1) ...\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jar to provide /usr/bin/jar (jar) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jarsigner to provide /usr/bin/jarsigner (jarsigner) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javac to provide /usr/bin/javac (javac) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javadoc to provide /usr/bin/javadoc (javadoc) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javap to provide /usr/bin/javap (javap) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jcmd to provide /usr/bin/jcmd (jcmd) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdb to provide /usr/bin/jdb (jdb) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeprscan to provide /usr/bin/jdeprscan (jdeprscan) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeps to provide /usr/bin/jdeps (jdeps) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jfr to provide /usr/bin/jfr (jfr) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jimage to provide /usr/bin/jimage (jimage) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jinfo to provide /usr/bin/jinfo (jinfo) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jlink to provide /usr/bin/jlink (jlink) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jmap to provide /usr/bin/jmap (jmap) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jmod to provide /usr/bin/jmod (jmod) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jps to provide /usr/bin/jps (jps) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jrunscript to provide /usr/bin/jrunscript (jrunscript) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jshell to provide /usr/bin/jshell (jshell) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstack to provide /usr/bin/jstack (jstack) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstat to provide /usr/bin/jstat (jstat) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstatd to provide /usr/bin/jstatd (jstatd) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/serialver to provide /usr/bin/serialver (serialver) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jhsdb to provide /usr/bin/jhsdb (jhsdb) in auto mode\n","Setting up x11-utils (7.7+5build2) ...\n","Setting up libatk-wrapper-java (0.38.0-5build1) ...\n","Setting up openjdk-17-jdk:amd64 (17.0.14+7-1~22.04.1) ...\n","update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jconsole to provide /usr/bin/jconsole (jconsole) in auto mode\n","Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n","Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n","Processing triggers for hicolor-icon-theme (0.17-2) ...\n","Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n","\n","Processing triggers for man-db (2.10.2-1) ...\n","Processing triggers for mailcap (3.70+nmu1ubuntu1) ...\n"]}],"source":["import os\n","# Update package list and install necessary tools\n","!apt-get update && apt-get install\n","!apt install -y fastqc samtools star bedtools wget unzip fastp\n","\n","\n","# Install Python libraries for scRNA-seq analysis\n","!pip install multiqc scanpy macs2 pybedtools deeptools biopython anndata matplotlib seaborn numpy pandas scipy scikit-learn umap-learn leidenalg cutadapt\n","# Install MACS2\n","!pip install macs2\n","\n","# Install SRA Toolkit for downloading SRA datasets\n","!wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz\n","!tar -xvzf sratoolkit.current-ubuntu64.tar.gz\n","!mkdir -p /usr/local/bin/sratoolkit\n","!mv sratoolkit.* /usr/local/bin/sratoolkit\n","os.environ[\"PATH\"] += \":/usr/local/bin/sratoolkit/sratoolkit.3.2.0-ubuntu64/bin/\"\n","\n","# Check that cutadapt is installed\n","!pip install cutadapt\n","\n","# Install Trim Galore\n","!curl -fsSL https://github.com/FelixKrueger/TrimGalore/archive/0.6.10.tar.gz -o trim_galore.tar.gz\n","!tar xvzf trim_galore.tar.gz\n","!mv /content/TrimGalore-0.6.10 /usr/local/bin/TrimGalore-0.6.10\n","os.environ[\"PATH\"] += \":/usr/local/bin/TrimGalore-0.6.10/\"\n","\n","# Install PRINSEQ\n","!wget http://prinseq.sourceforge.net/distributions/prinseq-lite-0.20.4.tar.gz\n","!tar -xvzf prinseq-lite-0.20.4.tar.gz\n","!mv prinseq-lite-0.20.4 /usr/local/bin/prinseq\n","os.environ[\"PATH\"] += \":/usr/local/bin/prinseq\"\n","\n","\n","# Install rpy2 to enable using R-based Seurat in Python\n","!pip install rpy2\n","\n","# Install R and Seurat for additional single-cell analysis\n","!apt-get install -y r-base\n","# Update java for picard\n","!apt-get install openjdk-17-jdk -y"]},{"cell_type":"code","execution_count":2,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":4747,"status":"ok","timestamp":1741549137988,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"ESIihJcXx4gx","outputId":"ba9a83a2-a361-41fc-8878-f4635ad05255"},"outputs":[{"output_type":"stream","name":"stdout","text":["--2025-03-09 19:38:52--  https://github.com/broadinstitute/picard/releases/download/2.27.4/picard.jar\n","Resolving github.com (github.com)... 20.27.177.113\n","Connecting to github.com (github.com)|20.27.177.113|:443... connected.\n","HTTP request sent, awaiting response... 302 Found\n","Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/839cd9dd-e7dc-4c29-ab0d-6ab4bdcded4e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250309%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250309T193853Z&X-Amz-Expires=300&X-Amz-Signature=6aa478240b5b461e6b4ba1e82a0acebd88813b442a4bc394b0616410bafa9324&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream [following]\n","--2025-03-09 19:38:53--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/839cd9dd-e7dc-4c29-ab0d-6ab4bdcded4e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250309%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250309T193853Z&X-Amz-Expires=300&X-Amz-Signature=6aa478240b5b461e6b4ba1e82a0acebd88813b442a4bc394b0616410bafa9324&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream\n","Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n","Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 17544754 (17M) [application/octet-stream]\n","Saving to: ‘/usr/local/bin/picard.jar’\n","\n","/usr/local/bin/pica 100%[===================>]  16.73M  5.96MB/s    in 2.8s    \n","\n","2025-03-09 19:38:57 (5.96 MB/s) - ‘/usr/local/bin/picard.jar’ saved [17544754/17544754]\n","\n"]}],"source":["# Install Picard (if not installed)\n","!wget https://github.com/broadinstitute/picard/releases/download/2.27.4/picard.jar -O /usr/local/bin/picard.jar"]},{"cell_type":"code","source":["!git clone https://github.com/alexdobin/STAR.git"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"FWrlU87pTiCi","executionInfo":{"status":"ok","timestamp":1741549175785,"user_tz":240,"elapsed":37810,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"57f9d472-c33b-4a4c-aafb-ceaa02b510ea"},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'STAR'...\n","remote: Enumerating objects: 11104, done.\u001b[K\n","remote: Counting objects: 100% (1143/1143), done.\u001b[K\n","remote: Compressing objects: 100% (372/372), done.\u001b[K\n","remote: Total 11104 (delta 777), reused 1081 (delta 720), pack-reused 9961 (from 1)\u001b[K\n","Receiving objects: 100% (11104/11104), 525.38 MiB | 22.77 MiB/s, done.\n","Resolving deltas: 100% (7853/7853), done.\n"]}]},{"cell_type":"code","source":["%cd STAR/source\n","!make STAR"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"8N-G7rk6TqhM","executionInfo":{"status":"ok","timestamp":1741549418574,"user_tz":240,"elapsed":242802,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"6fb084df-32cc-4646-a360-3e23c923afc9"},"execution_count":4,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/STAR/source\n","xxd -i parametersDefault > parametersDefault.xxd\n","make -C htslib lib-static\n","make[1]: Entering directory '/content/STAR/source/htslib'\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o kfunc.o kfunc.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o knetfile.o knetfile.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o kstring.o kstring.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o bgzf.o bgzf.c\n","\u001b[01m\u001b[Kbgzf.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kbgzf_write\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kbgzf.c:733:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n","  733 |     \u001b[01;35m\u001b[Kif\u001b[m\u001b[K ( !fp->is_compressed )\n","      |     \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n","\u001b[01m\u001b[Kbgzf.c:736:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n","  736 |         \u001b[01;36m\u001b[Kconst\u001b[m\u001b[K uint8_t *input = (const uint8_t*)data;\n","      |         \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o faidx.o faidx.c\n","\u001b[01m\u001b[Kfaidx.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kfai_load\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kfaidx.c:265:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kelse\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n","  265 |     \u001b[01;35m\u001b[Kelse\u001b[m\u001b[K\n","      |     \u001b[01;35m\u001b[K^~~~\u001b[m\u001b[K\n","\u001b[01m\u001b[Kfaidx.c:268:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kelse\u001b[m\u001b[K’\n","  268 |         \u001b[01;36m\u001b[Kif\u001b[m\u001b[K (fp == 0) {\n","      |         \u001b[01;36m\u001b[K^~\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hfile.o hfile.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hfile_net.o hfile_net.c\n","echo '#define HTS_VERSION \"0.0.1\"' > version.h\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hts.o hts.c\n","\u001b[01m\u001b[Khts.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Khts_idx_init\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Khts.c:522:63:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Koverflow in conversion from ‘\u001b[01m\u001b[Kuint32_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kunsigned int\u001b[m\u001b[K’} to ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ changes value from ‘\u001b[01m\u001b[Kidx->z.last_bin = 4294967295\u001b[m\u001b[K’ to ‘\u001b[01m\u001b[K-1\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverflow\u0007-Woverflow\u001b]8;;\u0007\u001b[m\u001b[K]\n","  522 |         idx->z.save_bin = idx->z.save_tid = idx->z.last_tid = \u001b[01;35m\u001b[Kidx\u001b[m\u001b[K->z.last_bin = 0xffffffffu;\n","      |                                                               \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o sam.o sam.c\n","In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n","                 from \u001b[01m\u001b[Ksam.c:3\u001b[m\u001b[K:\n","In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n","    inlined from ‘\u001b[01m\u001b[Kbam_hdr_write\u001b[m\u001b[K’ at \u001b[01m\u001b[Ksam.c:130:2\u001b[m\u001b[K:\n","\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying 4 bytes from a string of the same length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n","   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n","      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n","      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o synced_bcf_reader.o synced_bcf_reader.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcf_sweep.o vcf_sweep.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o tbx.o tbx.c\n","\u001b[01m\u001b[Ktbx.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ktbx_seqnames\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Ktbx.c:285:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kfor\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n","  285 |     \u001b[01;35m\u001b[Kfor\u001b[m\u001b[K (tid=0; tid<m; tid++)\n","      |     \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n","\u001b[01m\u001b[Ktbx.c:287:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kfor\u001b[m\u001b[K’\n","  287 |         \u001b[01;36m\u001b[K*\u001b[m\u001b[Kn = m;\n","      |         \u001b[01;36m\u001b[K^\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcf.o vcf.c\n","\u001b[01m\u001b[Kvcf.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kvcf_format\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kvcf.c:1909:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n"," 1909 |                         \u001b[01;35m\u001b[Kif\u001b[m\u001b[K ( !first ) kputc(';', s); first = 0;\n","      |                         \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n","\u001b[01m\u001b[Kvcf.c:1909:54:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n"," 1909 |                         if ( !first ) kputc(';', s); \u001b[01;36m\u001b[Kfirst\u001b[m\u001b[K = 0;\n","      |                                                      \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n","\u001b[01m\u001b[Kvcf.c:1948:21:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n"," 1948 |                     \u001b[01;35m\u001b[Kif\u001b[m\u001b[K (!first) kputc(':', s); first = 0;\n","      |                     \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n","\u001b[01m\u001b[Kvcf.c:1948:48:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n"," 1948 |                     if (!first) kputc(':', s); \u001b[01;36m\u001b[Kfirst\u001b[m\u001b[K = 0;\n","      |                                                \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcfutils.o vcfutils.c\n","In file included from \u001b[01m\u001b[Khtslib/vcf.h:12\u001b[m\u001b[K,\n","                 from \u001b[01m\u001b[Khtslib/vcfutils.h:8\u001b[m\u001b[K,\n","                 from \u001b[01m\u001b[Kvcfutils.c:1\u001b[m\u001b[K:\n","\u001b[01m\u001b[Kvcfutils.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kbcf_remove_alleles\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kvcfutils.c:291:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n","  291 |                 assert( \u001b[01;35m\u001b[Kn\u001b[m\u001b[K=nG_ori );\n","      |                         \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_codecs.o cram/cram_codecs.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_decode.o cram/cram_decode.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_encode.o cram/cram_encode.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_index.o cram/cram_index.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_io.o cram/cram_io.c\n","\u001b[01m\u001b[Kcram/cram_io.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kcram/cram_io.c:1529:34:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K.tmp_\u001b[m\u001b[K’ directive writing 5 bytes into a region of size between 1 and 4096 [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wformat-overflow=\u0007-Wformat-overflow=\u001b]8;;\u0007\u001b[m\u001b[K]\n"," 1529 |             sprintf(path_tmp, \"%s\u001b[01;35m\u001b[K.tmp_\u001b[m\u001b[K%d\", path, /*getpid(),*/ i);\n","      |                                  \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n","\u001b[01m\u001b[Kcram/cram_io.c:1529:31:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdirective argument in the range [0, 2147483647]\n"," 1529 |             sprintf(path_tmp, \u001b[01;36m\u001b[K\"%s.tmp_%d\"\u001b[m\u001b[K, path, /*getpid(),*/ i);\n","      |                               \u001b[01;36m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n","In file included from \u001b[01m\u001b[K/usr/include/stdio.h:894\u001b[m\u001b[K,\n","                 from \u001b[01m\u001b[Kcram/cram_io.c:51\u001b[m\u001b[K:\n","\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 7 and 4111 bytes into a destination of size 4096\n","   38 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n","      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","   39 | \u001b[01;36m\u001b[K                                  __glibc_objsize (__s), __fmt,\u001b[m\u001b[K\n","      |                                   \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","   40 | \u001b[01;36m\u001b[K                                  __va_arg_pack ())\u001b[m\u001b[K;\n","      |                                   \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n","                 from \u001b[01m\u001b[Kcram/cram_io.c:55\u001b[m\u001b[K:\n","In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n","    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2917:6\u001b[m\u001b[K,\n","    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2987:3\u001b[m\u001b[K:\n","\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n","   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n","      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n","      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n","    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2909:2\u001b[m\u001b[K,\n","    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2987:3\u001b[m\u001b[K:\n","\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n","   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n","      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n","      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n","    inlined from ‘\u001b[01m\u001b[Kcram_dopen\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3291:2\u001b[m\u001b[K:\n","\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 20 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n","   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n","      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n","      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_samtools.o cram/cram_samtools.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_stats.o cram/cram_stats.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/files.o cram/files.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/mFILE.o cram/mFILE.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/md5.o cram/md5.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/open_trace_file.o cram/open_trace_file.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/sam_header.o cram/sam_header.c\n","In file included from \u001b[01m\u001b[Kcram/sam_header.c:36\u001b[m\u001b[K:\n","\u001b[01m\u001b[Kcram/sam_header.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ksam_hdr_add_lines\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kcram/sam_header.c:328:20:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n","  328 |             assert(\u001b[01;35m\u001b[Kp\u001b[m\u001b[K->next = t);\n","      |                    \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n","\u001b[01m\u001b[Kcram/sam_header.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ksam_hdr_vadd\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kcram/sam_header.c:449:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n","  449 |         assert(\u001b[01;35m\u001b[Kp\u001b[m\u001b[K->next = t);\n","      |                \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/string_alloc.o cram/string_alloc.c\n","In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n","                 from \u001b[01m\u001b[Kcram/string_alloc.c:39\u001b[m\u001b[K:\n","In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n","    inlined from ‘\u001b[01m\u001b[Kstring_ndup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:149:5\u001b[m\u001b[K,\n","    inlined from ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:141:12\u001b[m\u001b[K:\n","\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying as many bytes from a string as its length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n","   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n","      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n","      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","\u001b[01m\u001b[Kcram/string_alloc.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kcram/string_alloc.c:141:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Klength computed here\n","  141 |     return \u001b[01;36m\u001b[Kstring_ndup(a_str, instr, strlen(instr))\u001b[m\u001b[K;\n","      |            \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/thread_pool.o cram/thread_pool.c\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/vlen.o cram/vlen.c\n","\u001b[01m\u001b[Kcram/vlen.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kvflen\u001b[m\u001b[K’:\n","\u001b[01m\u001b[Kcram/vlen.c:121:9:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Ki\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n","  121 |     int \u001b[01;35m\u001b[Ki\u001b[m\u001b[K;\n","      |         \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n","cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/zfio.o cram/zfio.c\n","ar -rc libhts.a kfunc.o knetfile.o kstring.o bgzf.o faidx.o hfile.o hfile_net.o hts.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/md5.o cram/open_trace_file.o cram/pooled_alloc.o cram/sam_header.o cram/string_alloc.o cram/thread_pool.o cram/vlen.o cram/zfio.o\n","ranlib libhts.a\n","make[1]: Leaving directory '/content/STAR/source/htslib'\n","echo BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp funPrimaryAlignMark.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp systemFunctions.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c\n","BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp funPrimaryAlignMark.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp systemFunctions.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c\n","'rm' -f ./Depend.list\n","g++ -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:39:35+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -MM BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp funPrimaryAlignMark.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp systemFunctions.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c parametersDefault.xxd htslib >> Depend.list\n","\u001b[01m\u001b[Kg++:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[KparametersDefault.xxd: linker input file unused because linking not done\n","\u001b[01m\u001b[Kg++:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Khtslib: linker input file unused because linking not done\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra systemFunctions.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra funPrimaryAlignMark.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_collapseUMI_Graph.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_collapseUMIall.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ParametersClip_initialize.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ClipMate_clip.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ClipCR4.cpp\n","cd opal && \\\n","g++ -c -I./ -std=c++11   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra  -mavx2 opal.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ClipMate_clipChunk.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ClipMate_initialize.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_loadRawMatrix.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_emptyDrops_CR.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra soloInputFeatureUMI.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_countSmartSeq.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_redistributeReadsByCB.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_quantTranscript.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_sumThreads.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_countVelocyto.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_countCBgeneUMI.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcriptome_classifyAlign.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_cellFiltering.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_statsOutput.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra bamSortByCoordinate.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloBarcode.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ParametersSolo.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloRead.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloRead_record.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloReadBarcode.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloReadBarcode_getCBandUMI.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloBarcode_extractBarcode.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloReadFeature.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloReadFeature_record.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloReadFeature_inputRecords.cpp\n","\u001b[01m\u001b[KSoloReadFeature_inputRecords.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid SoloReadFeature::inputRecords(unsigned int**, unsigned int, std::vector<unsigned int>&, std::vector<readInfoStruct>&, SoloReadFlagClass&, std::vector<unsigned int>&, std::vector<unsigned int>&)\u001b[m\u001b[K’:\n","\u001b[01m\u001b[KSoloReadFeature_inputRecords.cpp:21:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kunused variable ‘\u001b[01m\u001b[KnReadsIn\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable\u0007-Wunused-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n","   21 |     uint64 \u001b[01;35m\u001b[KnReadsIn\u001b[m\u001b[K = 0;\n","      |            \u001b[01;35m\u001b[K^~~~~~~~\u001b[m\u001b[K\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Solo.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature.cpp\n","In file included from \u001b[01m\u001b[KSoloFeature.cpp:1\u001b[m\u001b[K:\n","\u001b[01m\u001b[KSoloFeature.h:\u001b[m\u001b[K In constructor ‘\u001b[01m\u001b[KSoloFeature::SoloFeature(Parameters&, ReadAlignChunk**, Transcriptome&, int, SoloReadBarcode*, SoloFeature**)\u001b[m\u001b[K’:\n","\u001b[01m\u001b[KSoloFeature.h:36:17:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[KSoloFeature::featureType\u001b[m\u001b[K’ will be initialized after [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wreorder\u0007-Wreorder\u001b]8;;\u0007\u001b[m\u001b[K]\n","   36 |     const int32 \u001b[01;35m\u001b[KfeatureType\u001b[m\u001b[K;\n","      |                 \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n","\u001b[01m\u001b[KSoloFeature.h:24:19:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K  ‘\u001b[01m\u001b[KSoloFeature** SoloFeature::soloFeatAll\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wreorder\u0007-Wreorder\u001b]8;;\u0007\u001b[m\u001b[K]\n","   24 |     SoloFeature **\u001b[01;35m\u001b[KsoloFeatAll\u001b[m\u001b[K;\n","      |                   \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n","\u001b[01m\u001b[KSoloFeature.cpp:4:1:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K  when initialized here [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wreorder\u0007-Wreorder\u001b]8;;\u0007\u001b[m\u001b[K]\n","    4 | \u001b[01;35m\u001b[KSoloFeature\u001b[m\u001b[K::SoloFeature(Parameters &Pin, ReadAlignChunk **RAchunk, Transcriptome &inTrans, int32 feTy, SoloReadBarcode *readBarSumIn, SoloFeature **soloFeatAll)\n","      | \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_outputResults.cpp\n","\u001b[01m\u001b[KSoloFeature_outputResults.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid SoloFeature::outputResults(bool, std::string)\u001b[m\u001b[K’:\n","\u001b[01m\u001b[KSoloFeature_outputResults.cpp:51:23:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* getcwd(char*, size_t)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n","   51 |                 \u001b[01;35m\u001b[Kgetcwd(cwd,4096)\u001b[m\u001b[K;\n","      |                 \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~\u001b[m\u001b[K\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_processRecords.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SoloFeature_addBAMtags.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_transformGenome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Genome_transformGenome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcript_convertGenomeCigar.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra twoPassRunPass1.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra samHeaders.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Genome_genomeLoad.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Genome_genomeOutLoad.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcript_transformGenome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_outputSpliceGraphSAM.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_mapOneReadSpliceGraph.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SpliceGraph.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SpliceGraph_swScoreSpliced.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SpliceGraph_swTraceBack.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SpliceGraph_findSuperTr.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra sjAlignSplit.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra GTF.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra GTF_transcriptGeneSJ.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra GTF_superTranscript.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SuperTranscriptome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_outputAlignments.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra STAR.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SharedMemory.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra PackedArray.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SuffixArrayFuns.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Parameters.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Parameters_samAttributes.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra InOutStreams.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra SequenceFuns.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Genome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ParametersGenome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Stats.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcript.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcript_alignScore.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcript_generateCigarP.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Chain.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcript_variationAdjust.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Variation.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_waspMap.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_storeAligns.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_stitchPieces.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_multMapSelect.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_mapOneRead.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra readLoad.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlignChunk.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlignChunk_processChunks.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlignChunk_mapChunk.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra OutSJ.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra outputSJ.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra blocksOverlap.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ThreadControl.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra sysRemoveDir.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_maxMappableLength2strands.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra binarySearch2.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_outputTranscriptSAM.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_outputTranscriptSJ.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_outputTranscriptCIGARp.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_createExtendWindowsWithAlign.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_assignAlignToWindow.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_oneRead.cpp\n","\u001b[01m\u001b[KReadAlign_oneRead.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kint ReadAlign::oneRead()\u001b[m\u001b[K’:\n","\u001b[01m\u001b[KReadAlign_oneRead.cpp:23:21:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K*readStatus[0]\u001b[m\u001b[K’ may be used uninitialized [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmaybe-uninitialized\u0007-Wmaybe-uninitialized\u001b]8;;\u0007\u001b[m\u001b[K]\n","   23 |     if (\u001b[01;35m\u001b[KreadStatus[0]\u001b[m\u001b[K==-1) {//finished with the stream\n","      |         \u001b[01;35m\u001b[K~~~~~~~~~~~~^\u001b[m\u001b[K\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_stitchWindowSeeds.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_peOverlapMergeMap.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_mappedFilter.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ParametersChimeric_initialize.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_chimericDetection.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_chimericDetectionOld.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_chimericDetectionOldOutput.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ChimericDetection.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ChimericDetection_chimericDetectionMult.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_chimericDetectionPEmerged.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra stitchWindowAligns.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra extendAlign.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra stitchAlignToTranscript.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ChimericAlign_chimericJunctionOutput.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ChimericAlign_chimericBAMoutput.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ChimericAlign_chimericStitching.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Genome_genomeGenerate.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra genomeParametersWrite.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra genomeScanFastaFiles.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra genomeSAindex.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Genome_insertSequences.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra insertSeqSA.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra funCompareUintAndSuffixes.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra funCompareUintAndSuffixesMemcmp.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra TimeFunctions.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ErrorWarning.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra streamFuns.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra stringSubstituteAll.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcriptome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcriptome_quantAlign.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcriptome_geneFullAlignOverlap.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_quantTranscriptome.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Quantifications.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Transcriptome_geneCountsAddAlign.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra sjdbLoadFromFiles.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra sjdbLoadFromStream.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra sjdbPrepare.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra sjdbBuildIndex.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra sjdbInsertJunctions.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra mapThreadsSpawn.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Parameters_readFilesInit.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra Parameters_readSAMheader.cpp\n","\u001b[01m\u001b[KParameters_readSAMheader.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid Parameters::readSAMheader(std::string, std::vector<std::__cxx11::basic_string<char> >)\u001b[m\u001b[K’:\n","\u001b[01m\u001b[KParameters_readSAMheader.cpp:32:15:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint system(const char*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n","   32 |         \u001b[01;35m\u001b[Ksystem(com1.c_str())\u001b[m\u001b[K;\n","      |         \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~~~~~\u001b[m\u001b[K\n","g++ -c  -pipe -Wall -Wextra -O3 bam_cat.c\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra serviceFuns.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra BAMoutput.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra BAMfunctions.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra ReadAlign_alignBAM.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra BAMbinSortByCoordinate.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra signalFromBAM.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra bamRemoveDuplicates.cpp\n","g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra BAMbinSortUnmapped.cpp\n","g++ -o STAR  -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"On branch master ; commit b1edc1208d91a53bf40ebae8669f71d50b994851 ; diff files: \"' -pipe -Wall -Wextra systemFunctions.o funPrimaryAlignMark.o SoloFeature_collapseUMI_Graph.o SoloFeature_collapseUMIall.o ParametersClip_initialize.o ClipMate_clip.o ClipCR4.o opal/opal.o ClipMate_clipChunk.o ClipMate_initialize.o SoloFeature_loadRawMatrix.o SoloFeature_emptyDrops_CR.o soloInputFeatureUMI.o SoloFeature_countSmartSeq.o SoloFeature_redistributeReadsByCB.o SoloFeature_quantTranscript.o SoloFeature_sumThreads.o SoloFeature_countVelocyto.o SoloFeature_countCBgeneUMI.o Transcriptome_classifyAlign.o Transcriptome_geneFullAlignOverlap_ExonOverIntron.o Transcriptome_alignExonOverlap.cpp SoloFeature_cellFiltering.o SoloFeature_statsOutput.o bamSortByCoordinate.o SoloBarcode.o ParametersSolo.o SoloRead.o SoloRead_record.o SoloReadBarcode.o SoloReadBarcode_getCBandUMI.o SoloBarcode_extractBarcode.o SoloReadFeature.o SoloReadFeature_record.o SoloReadFeature_inputRecords.o Solo.o SoloFeature.o SoloFeature_outputResults.o SoloFeature_processRecords.o SoloFeature_addBAMtags.o ReadAlign_transformGenome.o Genome_transformGenome.o Transcript_convertGenomeCigar.o twoPassRunPass1.o samHeaders.o Genome_genomeLoad.o Genome_genomeOutLoad.o Transcript_transformGenome.o ReadAlign_outputSpliceGraphSAM.o ReadAlign_mapOneReadSpliceGraph.o SpliceGraph.o SpliceGraph_swScoreSpliced.o SpliceGraph_swTraceBack.o SpliceGraph_findSuperTr.o sjAlignSplit.o GTF.o GTF_transcriptGeneSJ.o GTF_superTranscript.o SuperTranscriptome.o ReadAlign_outputAlignments.o ReadAlign.o STAR.o SharedMemory.o PackedArray.o SuffixArrayFuns.o Parameters.o Parameters_samAttributes.o InOutStreams.o SequenceFuns.o Genome.o ParametersGenome.o Stats.o Transcript.o Transcript_alignScore.o Transcript_generateCigarP.o Chain.o Transcript_variationAdjust.o Variation.o ReadAlign_waspMap.o ReadAlign_storeAligns.o ReadAlign_stitchPieces.o ReadAlign_multMapSelect.o ReadAlign_mapOneRead.o readLoad.o ReadAlignChunk.o ReadAlignChunk_processChunks.o ReadAlignChunk_mapChunk.o OutSJ.o outputSJ.o blocksOverlap.o ThreadControl.o sysRemoveDir.o ReadAlign_maxMappableLength2strands.o binarySearch2.o ReadAlign_outputTranscriptSAM.o ReadAlign_outputTranscriptSJ.o ReadAlign_outputTranscriptCIGARp.o ReadAlign_calcCIGAR.cpp ReadAlign_createExtendWindowsWithAlign.o ReadAlign_assignAlignToWindow.o ReadAlign_oneRead.o ReadAlign_stitchWindowSeeds.o ReadAlign_peOverlapMergeMap.o ReadAlign_mappedFilter.o ParametersChimeric_initialize.o ReadAlign_chimericDetection.o ReadAlign_chimericDetectionOld.o ReadAlign_chimericDetectionOldOutput.o ChimericDetection.o ChimericDetection_chimericDetectionMult.o ReadAlign_chimericDetectionPEmerged.o stitchWindowAligns.o extendAlign.o stitchAlignToTranscript.o ChimericSegment.cpp ChimericAlign.cpp ChimericAlign_chimericJunctionOutput.o ChimericAlign_chimericBAMoutput.o ChimericAlign_chimericStitching.o Genome_genomeGenerate.o genomeParametersWrite.o genomeScanFastaFiles.o genomeSAindex.o Genome_insertSequences.o insertSeqSA.o funCompareUintAndSuffixes.o funCompareUintAndSuffixesMemcmp.o TimeFunctions.o ErrorWarning.o streamFuns.o stringSubstituteAll.o Transcriptome.o Transcriptome_quantAlign.o Transcriptome_geneFullAlignOverlap.o ReadAlign_quantTranscriptome.o Quantifications.o Transcriptome_geneCountsAddAlign.o sjdbLoadFromFiles.o sjdbLoadFromStream.o sjdbPrepare.o sjdbBuildIndex.o sjdbInsertJunctions.o mapThreadsSpawn.o Parameters_readFilesInit.o Parameters_openReadsFiles.cpp Parameters_closeReadsFiles.cpp Parameters_readSAMheader.o bam_cat.o serviceFuns.o GlobalVariables.cpp BAMoutput.o BAMfunctions.o ReadAlign_alignBAM.o BAMbinSortByCoordinate.o signalFromBAM.o bamRemoveDuplicates.o BAMbinSortUnmapped.o  -pthread -Lhtslib -Bstatic -lhts -Bdynamic -lz \n","\u001b[01m\u001b[KParameters_openReadsFiles.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid Parameters::openReadsFiles()\u001b[m\u001b[K’:\n","\u001b[01m\u001b[KParameters_openReadsFiles.cpp:25:20:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kimate\u001b[m\u001b[K’ might be clobbered by ‘\u001b[01m\u001b[Klongjmp\u001b[m\u001b[K’ or ‘\u001b[01m\u001b[Kvfork\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wclobbered\u0007-Wclobbered\u001b]8;;\u0007\u001b[m\u001b[K]\n","   25 |          for (uint \u001b[01;35m\u001b[Kimate\u001b[m\u001b[K=0;imate<readFilesNames.size();imate++) {//open readIn files\n","      |                    \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n"]}]},{"cell_type":"code","source":["!cp /content/STAR/source/STAR /usr/local/bin/STAR\n","os.environ[\"PATH\"] += \":/usr/local/bin/STAR\""],"metadata":{"id":"FEPFoSSEWZlO","executionInfo":{"status":"ok","timestamp":1741549418575,"user_tz":240,"elapsed":40,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"execution_count":5,"outputs":[]},{"cell_type":"code","source":["%cd /content/"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"31hmNCUiVKXe","executionInfo":{"status":"ok","timestamp":1741549418575,"user_tz":240,"elapsed":31,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"a24db3b3-24ca-4ba7-ff2b-5bb2e4c25524"},"execution_count":6,"outputs":[{"output_type":"stream","name":"stdout","text":["/content\n"]}]},{"cell_type":"markdown","metadata":{"id":"xm-Weio7MQbK"},"source":["## **Step 1: Data Acquisition (Downloading the ATAC-seq Dataset)**\n","\n","Now that I have set up my Google Colab environment, I need to download the raw sequencing data (FASTQ files) from the NCBI Sequence Read Archive (SRA)."]},{"cell_type":"code","execution_count":39,"metadata":{"id":"a-xChB5EE9U1","executionInfo":{"status":"ok","timestamp":1741505625137,"user_tz":240,"elapsed":106,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"outputs":[],"source":["# Create a directory for the raw ATAC-seq data\n","!mkdir -p /content/ATACseq_RawData"]},{"cell_type":"code","execution_count":41,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":902499,"status":"ok","timestamp":1741506646509,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"o6zTwisZNgLC","outputId":"090ae8de-b8e3-48dc-96a6-fb6dc57269ac"},"outputs":[{"output_type":"stream","name":"stdout","text":["/content/ATACseq_RawData\n","2025-03-09T07:35:44 prefetch.3.2.0: 1) Resolving 'SRR891268'...\n","2025-03-09T07:35:45 prefetch.3.2.0: Current preference is set to retrieve SRA Normalized Format files with full base quality scores\n","2025-03-09T07:35:46 prefetch.3.2.0: 1) Downloading 'SRR891268'...\n","2025-03-09T07:35:46 prefetch.3.2.0:  SRA Normalized Format file is being retrieved\n","2025-03-09T07:35:46 prefetch.3.2.0:  Downloading via HTTPS...\n","2025-03-09T07:50:19 prefetch.3.2.0:  HTTPS download succeed\n","2025-03-09T07:50:46 prefetch.3.2.0:  'SRR891268' is valid: 11863867959 bytes were streamed from 11863863861\n","2025-03-09T07:50:46 prefetch.3.2.0: 1) 'SRR891268' was downloaded successfully\n"]}],"source":["# Change directory to ATACseq_RawData\n","%cd /content/ATACseq_RawData\n","\n","# Download fastq files (SRR891268)\n","!prefetch SRR891268"]},{"cell_type":"code","execution_count":42,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":836858,"status":"ok","timestamp":1741507678530,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"1XRSnz30QUEA","outputId":"855a88c5-ccc6-40de-83c1-6ee5a9df546b"},"outputs":[{"output_type":"stream","name":"stdout","text":["spots read      : 192,904,649\n","reads read      : 385,809,298\n","reads written   : 385,809,298\n"]}],"source":["# Convert SRA files to FASTQ format (paired-end reads)\n","!fasterq-dump --split-files --outdir /content/ATACseq_RawData SRR891268"]},{"cell_type":"code","execution_count":43,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":75,"status":"ok","timestamp":1741508710536,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"fThkV-dIft4y","outputId":"13afb0fd-a339-4fef-b5f2-c2184652a2d7"},"outputs":[{"output_type":"stream","name":"stdout","text":["/content\n"]}],"source":["%cd ../"]},{"cell_type":"code","execution_count":44,"metadata":{"id":"svHxw_DJNhzz","executionInfo":{"status":"ok","timestamp":1741508715146,"user_tz":240,"elapsed":1005,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"outputs":[],"source":["!rm -r /content/ATACseq_RawData/SRR891268"]},{"cell_type":"code","source":["!cp -r /content/ATACseq_RawData/ /content/drive/MyDrive/ATAC_Seq/"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"dawhRqJhAk9Y","executionInfo":{"status":"ok","timestamp":1741509292610,"user_tz":240,"elapsed":536195,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"1a8688cd-eea8-4e3c-dbd9-c5e7c3809ff0"},"execution_count":45,"outputs":[{"output_type":"stream","name":"stdout","text":["cp: error writing '/content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_2.fastq': No space left on device\n"]}]},{"cell_type":"markdown","metadata":{"id":"uux3ijud4rV2"},"source":["## **Step 2: Quality Control (QC) of ATAC-seq Data**\n","\n","Now that I have the raw FASTQ files, I must assess their quality before proceeding with further analysis. This is a crucial step because poor-quality reads, adapter contamination, or low-sequencing depth can negatively impact downstream results.\n","Why I We Need Quality Control?\n","\n","Quality control (QC) ensures that the sequencing reads are of high quality and suitable for analysis. It helps:\n","\n","    Identify low-quality bases that may affect alignment.\n","    Detect adapter contamination from sequencing library preparation.\n","    Check for overrepresented sequences, which may indicate PCR amplification bias.\n","    Assess GC content distribution, which should be uniform for a well-balanced sequencing run.\n","    Examine sequence duplication levels, which may indicate technical artifacts."]},{"cell_type":"code","execution_count":46,"metadata":{"id":"y8-OdRrzgECo","executionInfo":{"status":"ok","timestamp":1741509292622,"user_tz":240,"elapsed":5,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"outputs":[],"source":["# Create a directory for QC results\n","!mkdir -p /content/ATACseq_QC"]},{"cell_type":"code","execution_count":48,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":1680888,"status":"ok","timestamp":1741512698624,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"sgb07otIpyJG","outputId":"028c334d-44a5-412d-b09c-621e23988c98"},"outputs":[{"output_type":"stream","name":"stdout","text":["Started analysis of SRR891268_1.fastq\n","Approx 5% complete for SRR891268_1.fastq\n","Approx 10% complete for SRR891268_1.fastq\n","Approx 15% complete for SRR891268_1.fastq\n","Approx 20% complete for SRR891268_1.fastq\n","Approx 25% complete for SRR891268_1.fastq\n","Approx 30% complete for SRR891268_1.fastq\n","Approx 35% complete for SRR891268_1.fastq\n","Approx 40% complete for SRR891268_1.fastq\n","Approx 45% complete for SRR891268_1.fastq\n","Approx 50% complete for SRR891268_1.fastq\n","Approx 55% complete for SRR891268_1.fastq\n","Approx 60% complete for SRR891268_1.fastq\n","Approx 65% complete for SRR891268_1.fastq\n","Approx 70% complete for SRR891268_1.fastq\n","Approx 75% complete for SRR891268_1.fastq\n","Approx 80% complete for SRR891268_1.fastq\n","Approx 85% complete for SRR891268_1.fastq\n","Approx 90% complete for SRR891268_1.fastq\n","Approx 95% complete for SRR891268_1.fastq\n","Analysis complete for SRR891268_1.fastq\n","Started analysis of SRR891268_2.fastq\n","Approx 5% complete for SRR891268_2.fastq\n","Approx 10% complete for SRR891268_2.fastq\n","Approx 15% complete for SRR891268_2.fastq\n","Approx 20% complete for SRR891268_2.fastq\n","Approx 25% complete for SRR891268_2.fastq\n","Approx 30% complete for SRR891268_2.fastq\n","Approx 35% complete for SRR891268_2.fastq\n","Approx 40% complete for SRR891268_2.fastq\n","Approx 45% complete for SRR891268_2.fastq\n","Approx 50% complete for SRR891268_2.fastq\n","Approx 55% complete for SRR891268_2.fastq\n","Approx 60% complete for SRR891268_2.fastq\n","Approx 65% complete for SRR891268_2.fastq\n","Approx 70% complete for SRR891268_2.fastq\n","Approx 75% complete for SRR891268_2.fastq\n","Approx 80% complete for SRR891268_2.fastq\n","Approx 85% complete for SRR891268_2.fastq\n","Approx 90% complete for SRR891268_2.fastq\n","Approx 95% complete for SRR891268_2.fastq\n","Analysis complete for SRR891268_2.fastq\n"]}],"source":["# Run FastQC on all raw FASTQ files\n","!fastqc -o /content/ATACseq_QC /content/ATACseq_RawData/*.fastq"]},{"cell_type":"markdown","metadata":{"id":"cyRggEnD4vOM"},"source":["## **Step 3: Trimming and Filtering Reads (Handling Adapters & Sequence Duplications)**\n","\n","Since my ATAC-seq reads did not pass QC checks, I need to trim and filter the raw sequences to remove unwanted regions.\n","This will ensure high-quality data before alignment.\n","Why Do I Need Trimming?\n","\n","  **Adapter Removal**\n","\n","        My files contain Nextera Transposase Sequence adapters, which must be removed.\n","        Adapters are artificial sequences added during library preparation and should not be present in sequencing reads.\n","        If left untrimmed, they can interfere with alignment and lead to false peaks.\n","\n","  **Low-Quality Base Removal**\n","\n","        Sequencing errors accumulate at the 3' end of reads.\n","        We will trim low-quality bases with Q<20 (strict threshold).\n","\n","  **Handling Sequence Duplication**\n","  \n","        ATAC-seq has high duplication levels due to PCR amplification.\n","        I will remove exact duplicates while retaining biologically relevant duplicate reads."]},{"cell_type":"code","execution_count":49,"metadata":{"id":"fYTMMFdwp-NY","executionInfo":{"status":"ok","timestamp":1741513381227,"user_tz":240,"elapsed":118,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"outputs":[],"source":["# Create a directory for trimmed reads\n","!mkdir -p /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed"]},{"cell_type":"code","execution_count":56,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":2719304,"status":"ok","timestamp":1741520587835,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"Xck4tYTb5f-8","outputId":"e043d5ee-2a6f-451a-d9e3-c9f3dbb35c6b"},"outputs":[{"output_type":"stream","name":"stdout","text":["Path to Cutadapt set as: 'cutadapt' (default)\n","Cutadapt seems to be working fine (tested command 'cutadapt --version')\n","Cutadapt version: 5.0\n","Cutadapt seems to be using Python 3! Proceeding with multi-core enabled Cutadapt using 8 cores\n","pigz 2.6\n","Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n","\n","Proceeding with 'pigz -p 4' for decompression\n","To decrease CPU usage of decompression, please install 'igzip' and run again\n","\n","No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n","\n","Output will be written into the directory: /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/\n","Writing report to '/content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/SRR891268_1.fastq_trimming_report.txt'\n","\n","SUMMARISING RUN PARAMETERS\n","==========================\n","Input filename: /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_1.fastq\n","Trimming mode: paired-end\n","Trim Galore version: 0.6.10\n","Cutadapt version: 5.0\n","Python version: 3.11.11\n","Number of cores used for trimming: 8\n","Quality Phred score cutoff: 20\n","Quality encoding type selected: ASCII+33\n","Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; user defined)\n","Maximum trimming error rate: 0.1 (default)\n","Minimum required adapter overlap (stringency): 1 bp\n","Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n","Running FastQC on the data once trimming has completed\n","\n","Cutadapt seems to be fairly up-to-date (version 5.0). Setting -j 8\n","Writing final adapter and quality trimmed output to SRR891268_1_trimmed.fq\n","\n","\n","  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_1.fastq <<< \n","10000000 sequences processed\n","20000000 sequences processed\n","30000000 sequences processed\n","40000000 sequences processed\n","50000000 sequences processed\n","60000000 sequences processed\n","70000000 sequences processed\n","80000000 sequences processed\n","90000000 sequences processed\n","This is cutadapt 5.0 with Python 3.11.11\n","Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a CTGTCTCTTATA /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_1.fastq\n","Processing single-end reads on 8 cores ...\n","\n","=== Summary ===\n","\n","Total reads processed:              98,397,000\n","Reads with adapters:                39,136,643 (39.8%)\n","Reads written (passing filters):    98,397,000 (100.0%)\n","\n","Total basepairs processed: 4,919,850,000 bp\n","Quality-trimmed:              17,943,842 bp (0.4%)\n","Total written (filtered):  4,750,749,976 bp (96.6%)\n","\n","=== Adapter 1 ===\n","\n","Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 39136643 times\n","\n","Minimum overlap: 1\n","No. of allowed errors:\n","1-9 bp: 0; 10-12 bp: 1\n","\n","Bases preceding removed adapters:\n","  A: 20.1%\n","  C: 33.0%\n","  G: 23.1%\n","  T: 23.3%\n","  none/other: 0.5%\n","\n","Overview of removed sequences\n","length\tcount\texpect\tmax.err\terror counts\n","1\t19573224\t24599250.0\t0\t19573224\n","2\t6238099\t6149812.5\t0\t6238099\n","3\t2075997\t1537453.1\t0\t2075997\n","4\t1194135\t384363.3\t0\t1194135\n","5\t1036349\t96090.8\t0\t1036349\n","6\t1024628\t24022.7\t0\t1024628\n","7\t1016369\t6005.7\t0\t1016369\n","8\t1206776\t1501.4\t0\t1206776\n","9\t1179062\t375.4\t0\t1173414 5648\n","10\t986418\t93.8\t1\t977563 8855\n","11\t765536\t23.5\t1\t759515 6021\n","12\t573056\t5.9\t1\t565057 7999\n","13\t487876\t5.9\t1\t483419 4457\n","14\t434687\t5.9\t1\t430530 4157\n","15\t258646\t5.9\t1\t255824 2822\n","16\t136612\t5.9\t1\t134910 1702\n","17\t97762\t5.9\t1\t93498 4264\n","18\t73265\t5.9\t1\t71925 1340\n","19\t76073\t5.9\t1\t74146 1927\n","20\t79665\t5.9\t1\t77992 1673\n","21\t92277\t5.9\t1\t90855 1422\n","22\t82201\t5.9\t1\t77614 4587\n","23\t77390\t5.9\t1\t75926 1464\n","24\t113252\t5.9\t1\t84725 28527\n","25\t14351\t5.9\t1\t7383 6968\n","26\t1875\t5.9\t1\t1323 552\n","27\t8151\t5.9\t1\t957 7194\n","28\t903\t5.9\t1\t427 476\n","29\t5035\t5.9\t1\t857 4178\n","30\t1840\t5.9\t1\t341 1499\n","31\t798\t5.9\t1\t506 292\n","32\t4553\t5.9\t1\t531 4022\n","33\t655\t5.9\t1\t334 321\n","34\t13027\t5.9\t1\t157 12870\n","35\t1274\t5.9\t1\t204 1070\n","36\t1562\t5.9\t1\t341 1221\n","37\t6434\t5.9\t1\t810 5624\n","38\t970\t5.9\t1\t578 392\n","39\t1524\t5.9\t1\t679 845\n","40\t1730\t5.9\t1\t861 869\n","41\t2843\t5.9\t1\t2121 722\n","42\t5297\t5.9\t1\t2021 3276\n","43\t3577\t5.9\t1\t2832 745\n","44\t2039\t5.9\t1\t1730 309\n","45\t503\t5.9\t1\t354 149\n","46\t887\t5.9\t1\t525 362\n","47\t8492\t5.9\t1\t2413 6079\n","48\t4295\t5.9\t1\t3426 869\n","49\t4916\t5.9\t1\t2496 2420\n","50\t159757\t5.9\t1\t151767 7990\n","\n","RUN STATISTICS FOR INPUT FILE: /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_1.fastq\n","=============================================\n","98397000 sequences processed in total\n","The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n","\n","Writing report to '/content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/SRR891268_2.fastq_trimming_report.txt'\n","\n","SUMMARISING RUN PARAMETERS\n","==========================\n","Input filename: /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_2.fastq\n","Trimming mode: paired-end\n","Trim Galore version: 0.6.10\n","Cutadapt version: 5.0\n","Python version: 3.11.11\n","Number of cores used for trimming: 8\n","Quality Phred score cutoff: 20\n","Quality encoding type selected: ASCII+33\n","Adapter sequence: 'CTGTCTCTTATA' (Nextera Transposase sequence; user defined)\n","Maximum trimming error rate: 0.1 (default)\n","Minimum required adapter overlap (stringency): 1 bp\n","Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n","Running FastQC on the data once trimming has completed\n","\n","Cutadapt seems to be fairly up-to-date (version 5.0). Setting -j -j 8\n","Writing final adapter and quality trimmed output to SRR891268_2_trimmed.fq\n","\n","\n","  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'CTGTCTCTTATA' from file /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_2.fastq <<< \n","10000000 sequences processed\n","20000000 sequences processed\n","30000000 sequences processed\n","40000000 sequences processed\n","50000000 sequences processed\n","60000000 sequences processed\n","70000000 sequences processed\n","80000000 sequences processed\n","90000000 sequences processed\n","This is cutadapt 5.0 with Python 3.11.11\n","Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a CTGTCTCTTATA /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_2.fastq\n","Processing single-end reads on 8 cores ...\n","\n","=== Summary ===\n","\n","Total reads processed:              98,397,000\n","Reads with adapters:                38,790,748 (39.4%)\n","Reads written (passing filters):    98,397,000 (100.0%)\n","\n","Total basepairs processed: 4,919,850,000 bp\n","Quality-trimmed:              68,470,992 bp (1.4%)\n","Total written (filtered):  4,702,170,336 bp (95.6%)\n","\n","=== Adapter 1 ===\n","\n","Sequence: CTGTCTCTTATA; Type: regular 3'; Length: 12; Trimmed: 38790748 times\n","\n","Minimum overlap: 1\n","No. of allowed errors:\n","1-9 bp: 0; 10-12 bp: 1\n","\n","Bases preceding removed adapters:\n","  A: 20.0%\n","  C: 33.0%\n","  G: 23.1%\n","  T: 23.4%\n","  none/other: 0.5%\n","\n","Overview of removed sequences\n","length\tcount\texpect\tmax.err\terror counts\n","1\t19389888\t24599250.0\t0\t19389888\n","2\t6210870\t6149812.5\t0\t6210870\n","3\t2074861\t1537453.1\t0\t2074861\n","4\t1186571\t384363.3\t0\t1186571\n","5\t1027527\t96090.8\t0\t1027527\n","6\t1015200\t24022.7\t0\t1015200\n","7\t1005159\t6005.7\t0\t1005159\n","8\t1195333\t1501.4\t0\t1195333\n","9\t1177304\t375.4\t0\t1171772 5532\n","10\t969392\t93.8\t1\t961699 7693\n","11\t755554\t23.5\t1\t750401 5153\n","12\t566182\t5.9\t1\t559343 6839\n","13\t477034\t5.9\t1\t473477 3557\n","14\t426879\t5.9\t1\t423515 3364\n","15\t253345\t5.9\t1\t251070 2275\n","16\t135013\t5.9\t1\t133618 1395\n","17\t96868\t5.9\t1\t92672 4196\n","18\t72335\t5.9\t1\t71087 1248\n","19\t76577\t5.9\t1\t74476 2101\n","20\t77081\t5.9\t1\t75644 1437\n","21\t87470\t5.9\t1\t86322 1148\n","22\t81170\t5.9\t1\t77135 4035\n","23\t75454\t5.9\t1\t74241 1213\n","24\t108941\t5.9\t1\t85795 23146\n","25\t13477\t5.9\t1\t7296 6181\n","26\t1631\t5.9\t1\t1151 480\n","27\t6951\t5.9\t1\t721 6230\n","28\t1134\t5.9\t1\t653 481\n","29\t5228\t5.9\t1\t1092 4136\n","30\t1714\t5.9\t1\t332 1382\n","31\t965\t5.9\t1\t645 320\n","32\t4711\t5.9\t1\t855 3856\n","33\t760\t5.9\t1\t443 317\n","34\t11538\t5.9\t1\t560 10978\n","35\t1125\t5.9\t1\t186 939\n","36\t1383\t5.9\t1\t268 1115\n","37\t5451\t5.9\t1\t660 4791\n","38\t724\t5.9\t1\t296 428\n","39\t1076\t5.9\t1\t377 699\n","40\t1304\t5.9\t1\t457 847\n","41\t1305\t5.9\t1\t616 689\n","42\t4010\t5.9\t1\t708 3302\n","43\t1559\t5.9\t1\t813 746\n","44\t1892\t5.9\t1\t1530 362\n","45\t2420\t5.9\t1\t2112 308\n","46\t4385\t5.9\t1\t4025 360\n","47\t6994\t5.9\t1\t1508 5486\n","48\t4591\t5.9\t1\t3551 1040\n","49\t17328\t5.9\t1\t15051 2277\n","50\t145084\t5.9\t1\t138012 7072\n","\n","RUN STATISTICS FOR INPUT FILE: /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_2.fastq\n","=============================================\n","98397000 sequences processed in total\n","The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n","\n","Validate paired-end files SRR891268_1_trimmed.fq and SRR891268_2_trimmed.fq\n","file_1: SRR891268_1_trimmed.fq, file_2: SRR891268_2_trimmed.fq\n","\n","\n",">>>>> Now validing the length of the 2 paired-end infiles: SRR891268_1_trimmed.fq and SRR891268_2_trimmed.fq <<<<<\n","Writing validated paired-end Read 1 reads to SRR891268_1_val_1.fq\n","Writing validated paired-end Read 2 reads to SRR891268_2_val_2.fq\n","\n","Total number of sequences analysed: 98397000\n","\n","Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1156375 (1.18%)\n","\n","\n","  >>> Now running FastQC on the validated data SRR891268_1_val_1.fq<<<\n","\n","Started analysis of SRR891268_1_val_1.fq\n","Approx 5% complete for SRR891268_1_val_1.fq\n","Approx 10% complete for SRR891268_1_val_1.fq\n","Approx 15% complete for SRR891268_1_val_1.fq\n","Approx 20% complete for SRR891268_1_val_1.fq\n","Approx 25% complete for SRR891268_1_val_1.fq\n","Approx 30% complete for SRR891268_1_val_1.fq\n","Approx 35% complete for SRR891268_1_val_1.fq\n","Approx 40% complete for SRR891268_1_val_1.fq\n","Approx 45% complete for SRR891268_1_val_1.fq\n","Approx 50% complete for SRR891268_1_val_1.fq\n","Approx 55% complete for SRR891268_1_val_1.fq\n","Approx 60% complete for SRR891268_1_val_1.fq\n","Approx 65% complete for SRR891268_1_val_1.fq\n","Approx 70% complete for SRR891268_1_val_1.fq\n","Approx 75% complete for SRR891268_1_val_1.fq\n","Approx 80% complete for SRR891268_1_val_1.fq\n","Approx 85% complete for SRR891268_1_val_1.fq\n","Approx 90% complete for SRR891268_1_val_1.fq\n","Approx 95% complete for SRR891268_1_val_1.fq\n","Analysis complete for SRR891268_1_val_1.fq\n","\n","  >>> Now running FastQC on the validated data SRR891268_2_val_2.fq<<<\n","\n","Started analysis of SRR891268_2_val_2.fq\n","Approx 5% complete for SRR891268_2_val_2.fq\n","Approx 10% complete for SRR891268_2_val_2.fq\n","Approx 15% complete for SRR891268_2_val_2.fq\n","Approx 20% complete for SRR891268_2_val_2.fq\n","Approx 25% complete for SRR891268_2_val_2.fq\n","Approx 30% complete for SRR891268_2_val_2.fq\n","Approx 35% complete for SRR891268_2_val_2.fq\n","Approx 40% complete for SRR891268_2_val_2.fq\n","Approx 45% complete for SRR891268_2_val_2.fq\n","Approx 50% complete for SRR891268_2_val_2.fq\n","Approx 55% complete for SRR891268_2_val_2.fq\n","Approx 60% complete for SRR891268_2_val_2.fq\n","Approx 65% complete for SRR891268_2_val_2.fq\n","Approx 70% complete for SRR891268_2_val_2.fq\n","Approx 75% complete for SRR891268_2_val_2.fq\n","Approx 80% complete for SRR891268_2_val_2.fq\n","Approx 85% complete for SRR891268_2_val_2.fq\n","Approx 90% complete for SRR891268_2_val_2.fq\n","Approx 95% complete for SRR891268_2_val_2.fq\n","Analysis complete for SRR891268_2_val_2.fq\n","Deleting both intermediate output files SRR891268_1_trimmed.fq and SRR891268_2_trimmed.fq\n","\n","====================================================================================================\n","\n"]}],"source":["# Run Trim Galore with Nextera adapter removal\n","!trim_galore --paired --cores 8 --fastqc --nextera \\\n","  -o /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed \\\n","  /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_1.fastq /content/drive/MyDrive/ATAC_Seq/ATACseq_RawData/SRR891268_2.fastq"]},{"cell_type":"code","source":["!apt install fastp"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Nj-1-LmYdokq","executionInfo":{"status":"ok","timestamp":1741549708026,"user_tz":240,"elapsed":7395,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"d80db509-2625-4aad-bbb0-bc37e21c7cea"},"execution_count":9,"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","The following NEW packages will be installed:\n","  fastp\n","0 upgraded, 1 newly installed, 0 to remove and 33 not upgraded.\n","Need to get 193 kB of archives.\n","After this operation, 640 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastp amd64 0.20.1+dfsg-1 [193 kB]\n","Fetched 193 kB in 2s (127 kB/s)\n","Selecting previously unselected package fastp.\n","(Reading database ... 125889 files and directories currently installed.)\n","Preparing to unpack .../fastp_0.20.1+dfsg-1_amd64.deb ...\n","Unpacking fastp (0.20.1+dfsg-1) ...\n","Setting up fastp (0.20.1+dfsg-1) ...\n","Processing triggers for man-db (2.10.2-1) ...\n"]}]},{"cell_type":"code","execution_count":10,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":722058,"status":"ok","timestamp":1741550440415,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"jBS5GNcbzPDQ","outputId":"4bccb560-05a0-4437-e041-4e23fe11370d"},"outputs":[{"output_type":"stream","name":"stdout","text":["Read1 before filtering:\n","total reads: 97240625\n","total bases: 4707122343\n","Q20 bases: 4681649978(99.4589%)\n","Q30 bases: 4608999348(97.9154%)\n","\n","Read2 before filtering:\n","total reads: 97240625\n","total bases: 4693250062\n","Q20 bases: 4648096181(99.0379%)\n","Q30 bases: 4551970947(96.9897%)\n","\n","Read1 after filtering:\n","total reads: 97188534\n","total bases: 4704155276\n","Q20 bases: 4678743071(99.4598%)\n","Q30 bases: 4606200892(97.9177%)\n","\n","Read2 aftering filtering:\n","total reads: 97188534\n","total bases: 4690378086\n","Q20 bases: 4645534508(99.0439%)\n","Q30 bases: 4549604612(96.9987%)\n","\n","Filtering result:\n","reads passed filter: 194377068\n","reads failed due to low quality: 12396\n","reads failed due to too many N: 662\n","reads failed due to too short: 0\n","reads failed due to low complexity: 91124\n","reads with adapter trimmed: 156508\n","bases trimmed due to adapters: 1393450\n","\n","Duplication rate: 20.6462%\n","\n","Insert size peak (evaluated by paired-end reads): 42\n","\n","JSON report: /content/ATACseq_Trimmed/SRR30275166_fastp_report.json\n","HTML report: /content/ATACseq_Trimmed/SRR30275166_fastp_report.html\n","\n","fastp -i /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/SRR891268_1_val_1.fq -I /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/SRR891268_2_val_2.fq -o /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_1_filtered.fq -O /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_2_filtered.fq -y --complexity_threshold 30 --html /content/ATACseq_Trimmed/SRR30275166_fastp_report.html --json /content/ATACseq_Trimmed/SRR30275166_fastp_report.json \n","fastp v0.20.1, time used: 721 seconds\n"]}],"source":["!mkdir -p /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered\n","# Filter low-complexity reads\n","!fastp -i /ATACseq_Trimmed/SRR891268_1_val_1.fq \\\n","      -I /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/SRR891268_2_val_2.fq \\\n","      -o /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_1_filtered.fq \\\n","      -O /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_2_filtered.fq \\\n","      -y --complexity_threshold 30 \\\n","      --html /content/ATACseq_Trimmed/SRR30275166_fastp_report.html \\\n","      --json /content/ATACseq_Trimmed/SRR30275166_fastp_report.json/content/drive/MyDrive/ATAC_Seq"]},{"cell_type":"code","execution_count":11,"metadata":{"id":"iKHZT2tw0R8E","executionInfo":{"status":"ok","timestamp":1741550441850,"user_tz":240,"elapsed":13,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"outputs":[],"source":["# Create output directory for fastqc resukts\n","!mkdir -p /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/FastQC_Results"]},{"cell_type":"code","execution_count":14,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"executionInfo":{"elapsed":794240,"status":"ok","timestamp":1741551337454,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"hX0GTaiL0UWx","outputId":"099c464c-03af-4ca4-bd9c-431bfc36fd15"},"outputs":[{"output_type":"stream","name":"stdout","text":["Started analysis of SRR891268_1_filtered.fq\n","Approx 5% complete for SRR891268_1_filtered.fq\n","Approx 10% complete for SRR891268_1_filtered.fq\n","Approx 15% complete for SRR891268_1_filtered.fq\n","Approx 20% complete for SRR891268_1_filtered.fq\n","Approx 25% complete for SRR891268_1_filtered.fq\n","Approx 30% complete for SRR891268_1_filtered.fq\n","Approx 35% complete for SRR891268_1_filtered.fq\n","Approx 40% complete for SRR891268_1_filtered.fq\n","Approx 45% complete for SRR891268_1_filtered.fq\n","Approx 50% complete for SRR891268_1_filtered.fq\n","Approx 55% complete for SRR891268_1_filtered.fq\n","Approx 60% complete for SRR891268_1_filtered.fq\n","Approx 65% complete for SRR891268_1_filtered.fq\n","Approx 70% complete for SRR891268_1_filtered.fq\n","Approx 75% complete for SRR891268_1_filtered.fq\n","Approx 80% complete for SRR891268_1_filtered.fq\n","Approx 85% complete for SRR891268_1_filtered.fq\n","Approx 90% complete for SRR891268_1_filtered.fq\n","Approx 95% complete for SRR891268_1_filtered.fq\n","Analysis complete for SRR891268_1_filtered.fq\n","Started analysis of SRR891268_2_filtered.fq\n","Approx 5% complete for SRR891268_2_filtered.fq\n","Approx 10% complete for SRR891268_2_filtered.fq\n","Approx 15% complete for SRR891268_2_filtered.fq\n","Approx 20% complete for SRR891268_2_filtered.fq\n","Approx 25% complete for SRR891268_2_filtered.fq\n","Approx 30% complete for SRR891268_2_filtered.fq\n","Approx 35% complete for SRR891268_2_filtered.fq\n","Approx 40% complete for SRR891268_2_filtered.fq\n","Approx 45% complete for SRR891268_2_filtered.fq\n","Approx 50% complete for SRR891268_2_filtered.fq\n","Approx 55% complete for SRR891268_2_filtered.fq\n","Approx 60% complete for SRR891268_2_filtered.fq\n","Approx 65% complete for SRR891268_2_filtered.fq\n","Approx 70% complete for SRR891268_2_filtered.fq\n","Approx 75% complete for SRR891268_2_filtered.fq\n","Approx 80% complete for SRR891268_2_filtered.fq\n","Approx 85% complete for SRR891268_2_filtered.fq\n","Approx 90% complete for SRR891268_2_filtered.fq\n","Approx 95% complete for SRR891268_2_filtered.fq\n","Analysis complete for SRR891268_2_filtered.fq\n"]}],"source":["# Run FastQC for filtered reads\n","!fastqc /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_1_filtered.fq \\\n","        /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_2_filtered.fq \\\n","        -o /content/drive/MyDrive/ATAC_Seq/ATACseq_Trimmed/FastQC_Results"]},{"cell_type":"markdown","metadata":{"id":"EWq2qOv2rPNA"},"source":["## **Step 4: Read Alignment Using Bowtie2**\n","\n","Now that I have decided to proceed with Bowtie2, I will align the ATAC-seq reads to the human reference genome (GRCh38).\n","\n","Why is this step important?\n","\n","    ATAC-seq produces short DNA sequences that need to be mapped to the genome.\n","    Without alignment, we cannot identify open chromatin regions or regulatory elements.\n","    The output (BAM files) will be used for peak calling and visualization.\n","\n","Step-by-Step Execution\n","1. Download and Prepare the Reference Genome\n","\n","Before alignment, I need to download and index the human reference genome (GRCh38), which is required by Bowtie2.\n","\n","📌 Why do I need a reference genome?\n","\n","    The reference genome acts as a template to which reads are aligned.\n","    Bowtie2 needs a pre-indexed genome to speed up alignment.\n","\n","📌 Why use a pre-built Bowtie2 index?\n","\n","    Creating a Bowtie2 index from scratch is computationally expensive.\n","    Instead, I will download a pre-indexed GRCh38 genome, which saves time."]},{"cell_type":"code","execution_count":null,"metadata":{"executionInfo":{"elapsed":11,"status":"ok","timestamp":1739946486563,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":300},"id":"BKmkV78krv2R","colab":{"base_uri":"https://localhost:8080/"},"outputId":"9f56d9e6-d250-4d90-d778-3e7917728603"},"outputs":[{"output_type":"stream","name":"stdout","text":["rm: cannot remove '/content/ATACseq_Trimmed/FastQC_Results': No such file or directory\n"]}],"source":["#!rm -r /content/ATACseq_Trimmed/FastQC_Results"]},{"cell_type":"code","execution_count":15,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"b1E8-uJsz9T9","executionInfo":{"status":"ok","timestamp":1741558582767,"user_tz":240,"elapsed":5929741,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"21e68b37-1e82-48dc-eee7-79c7b310257e"},"outputs":[{"output_type":"stream","name":"stdout","text":["Downloading Human Genome (GRCh38)...\n","Downloading Gene Annotation (GTF)...\n","Indexing genome with STAR (This may take some time)...\n","STAR genome indexing completed!\n"]}],"source":["import os\n","\n","# Create directories for reference genome and index\n","os.makedirs(\"/content/drive/MyDrive/ATAC_Seq/reference_genome\", exist_ok=True)\n","os.makedirs(\"/content/drive/MyDrive/ATAC_Seq/STAR_index\", exist_ok=True)\n","\n","# Define paths\n","genome_fasta = \"/content/drive/MyDrive/ATAC_Seq/STAR_index/Homo_sapiens.GRCh38.dna.primary_assembly.fa\"\n","gtf_file = \"/content/drive/MyDrive/ATAC_Seq/STAR_index/Homo_sapiens.GRCh38.109.gtf\"\n","star_index_dir = \"/content/drive/MyDrive/ATAC_Seq/reference_genome\"\n","\n","# Download Human Genome (GRCh38) from Ensembl\n","print(\"Downloading Human Genome (GRCh38)...\")\n","os.system(f\"wget -O {genome_fasta}.gz ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\")\n","os.system(f\"gunzip {genome_fasta}.gz\")  # Decompress\n","\n","# Download Gene Annotation (GTF file) from Ensembl\n","print(\"Downloading Gene Annotation (GTF)...\")\n","os.system(f\"wget -O {gtf_file}.gz ftp://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz\")\n","os.system(f\"gunzip {gtf_file}.gz\")  # Decompress\n","\n","# Index the genome using STAR\n","print(\"Indexing genome with STAR (This may take some time)...\")\n","os.system(f\"\"\"\n","    STAR --runThreadN 8 \\\n","         --runMode genomeGenerate \\\n","         --genomeDir {star_index_dir} \\\n","         --genomeFastaFiles {genome_fasta} \\\n","         --sjdbGTFfile {gtf_file} \\\n","         --sjdbOverhang 100\n","\"\"\")\n","\n","print(\"STAR genome indexing completed!\")\n"]},{"cell_type":"code","execution_count":16,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":1285785,"status":"ok","timestamp":1741559868544,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"24V6keERyU_x","outputId":"cc620a05-6557-4b5c-eb3a-7cc3e666e29e"},"outputs":[{"output_type":"stream","name":"stdout","text":["\tSTAR --runThreadN 8 --genomeDir /content/drive/MyDrive/ATAC_Seq/reference_genome --readFilesIn /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_1_filtered.fq /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_2_filtered.fq --outFileNamePrefix /content/drive/MyDrive/ATAC_Seq/ATACseq_STAR_Aligned/SRR891268_ --outSAMtype BAM SortedByCoordinate --alignIntronMax 1\n","\tSTAR version: 2.7.11b   compiled: 2025-03-09T19:40:06+00:00 cdcdbd94fd8a:/content/STAR/source\n","Mar 09 22:16:22 ..... started STAR run\n","Mar 09 22:16:23 ..... loading genome\n","Mar 09 22:18:44 ..... started mapping\n","Mar 09 22:31:18 ..... finished mapping\n","Mar 09 22:31:21 ..... started sorting BAM\n","Mar 09 22:37:46 ..... finished successfully\n"]}],"source":["# Create output directory for aligned reads\n","!mkdir -p /content/drive/MyDrive/ATAC_Seq/ATACseq_STAR_Aligned\n","\n","# Align the reads\n","!STAR --runThreadN 8 \\\n","--genomeDir /content/drive/MyDrive/ATAC_Seq/reference_genome \\\n","--readFilesIn /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_1_filtered.fq /content/drive/MyDrive/ATAC_Seq/ATACseq_Filtered/SRR891268_2_filtered.fq \\\n","--outFileNamePrefix /content/drive/MyDrive/ATAC_Seq/ATACseq_STAR_Aligned/SRR891268_ \\\n","--outSAMtype BAM SortedByCoordinate \\\n","--alignIntronMax 1"]},{"cell_type":"code","execution_count":17,"metadata":{"executionInfo":{"elapsed":73032,"status":"ok","timestamp":1741561024999,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"},"user_tz":240},"id":"oZ-eBF5G4VF2"},"outputs":[],"source":["!cp /content/drive/MyDrive/ATAC_Seq/ATACseq_STAR_Aligned/SRR891268_Aligned.sortedByCoord.out.bam /content"]},{"cell_type":"code","source":["!samtools flagstat /content/SRR891268_Aligned.sortedByCoord.out.bam"],"metadata":{"id":"LsY63vuz9SoN","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1741564318752,"user_tz":240,"elapsed":99119,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"29b621ab-9705-4bcf-c0ba-6704eb82fc7d"},"execution_count":20,"outputs":[{"output_type":"stream","name":"stdout","text":["216288694 + 0 in total (QC-passed reads + QC-failed reads)\n","188385907 + 0 primary\n","27902787 + 0 secondary\n","0 + 0 supplementary\n","0 + 0 duplicates\n","0 + 0 primary duplicates\n","216288694 + 0 mapped (100.00% : N/A)\n","188385907 + 0 primary mapped (100.00% : N/A)\n","188385907 + 0 paired in sequencing\n","94210155 + 0 read1\n","94175752 + 0 read2\n","188344624 + 0 properly paired (99.98% : N/A)\n","188344624 + 0 with itself and mate mapped\n","41283 + 0 singletons (0.02% : N/A)\n","0 + 0 with mate mapped to a different chr\n","0 + 0 with mate mapped to a different chr (mapQ>=5)\n"]}]},{"cell_type":"code","source":["!samtools index /content/SRR891268_Aligned.sortedByCoord.out.bam"],"metadata":{"id":"N4HYwFzaUOB6","executionInfo":{"status":"ok","timestamp":1741564611661,"user_tz":240,"elapsed":98334,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"execution_count":21,"outputs":[]},{"cell_type":"code","source":["# Convert BAM to BED for visualization\n","!bedtools bamtobed -i /content/SRR891268_Aligned.sortedByCoord.out.bam > /content/SRR891268_reads.bed\n"],"metadata":{"id":"whhmO5uFWyd1","executionInfo":{"status":"ok","timestamp":1741565391227,"user_tz":240,"elapsed":445262,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"execution_count":25,"outputs":[]},{"cell_type":"code","source":["# Create output directory for peak calling results\n","!mkdir -p /content/ATACseq_Peaks"],"metadata":{"id":"dcEyZ65zW9w3","executionInfo":{"status":"ok","timestamp":1741575401885,"user_tz":240,"elapsed":80,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"execution_count":26,"outputs":[]},{"cell_type":"code","source":["# Run MACS2 peak calling\n","!macs2 callpeak -t /content/SRR891268_Aligned.sortedByCoord.out.bam \\\n","-f BAM \\\n","-g hs \\\n","-n SRR891268_ATAC_Peaks \\\n","--outdir /content/ATACseq_Peaks \\\n","--nomodel --shift -100 --extsize 200 \\\n","--keep-dup all \\\n","-B --SPMR\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"7FQRt3gj-6_N","executionInfo":{"status":"ok","timestamp":1741576271792,"user_tz":240,"elapsed":861795,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"b111f8e9-d502-4ee4-a59a-9de54034063c"},"execution_count":27,"outputs":[{"output_type":"stream","name":"stdout","text":["INFO  @ Mon, 10 Mar 2025 02:56:50: \n","# Command line: callpeak -t /content/SRR891268_Aligned.sortedByCoord.out.bam -f BAM -g hs -n SRR891268_ATAC_Peaks --outdir /content/ATACseq_Peaks --nomodel --shift -100 --extsize 200 --keep-dup all -B --SPMR\n","# ARGUMENTS LIST:\n","# name = SRR891268_ATAC_Peaks\n","# format = BAM\n","# ChIP-seq file = ['/content/SRR891268_Aligned.sortedByCoord.out.bam']\n","# control file = None\n","# effective genome size = 2.70e+09\n","# band width = 300\n","# model fold = [5, 50]\n","# qvalue cutoff = 5.00e-02\n","# The maximum gap between significant sites is assigned as the read length/tag size.\n","# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n","# Larger dataset will be scaled towards smaller dataset.\n","# Range for calculating regional lambda is: 10000 bps\n","# Broad region calling is off\n","# Paired-End mode is off\n","# MACS will save fragment pileup signal per million reads\n"," \n","INFO  @ Mon, 10 Mar 2025 02:56:50: #1 read tag files... \n","INFO  @ Mon, 10 Mar 2025 02:56:50: #1 read treatment tags... \n","INFO  @ Mon, 10 Mar 2025 02:56:55:  1000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:01:  2000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:08:  3000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:13:  4000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:17:  5000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:22:  6000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:26:  7000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:30:  8000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:35:  9000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:39:  10000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:43:  11000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:47:  12000000 \n","INFO  @ Mon, 10 Mar 2025 02:57:51:  13000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:00:  14000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:04:  15000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:09:  16000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:13:  17000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:17:  18000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:21:  19000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:26:  20000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:30:  21000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:34:  22000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:40:  23000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:44:  24000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:49:  25000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:53:  26000000 \n","INFO  @ Mon, 10 Mar 2025 02:58:57:  27000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:01:  28000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:05:  29000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:10:  30000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:15:  31000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:19:  32000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:23:  33000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:27:  34000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:31:  35000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:35:  36000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:39:  37000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:44:  38000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:48:  39000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:51:  40000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:55:  41000000 \n","INFO  @ Mon, 10 Mar 2025 02:59:58:  42000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:03:  43000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:07:  44000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:11:  45000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:15:  46000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:18:  47000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:21:  48000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:25:  49000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:28:  50000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:32:  51000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:36:  52000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:41:  53000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:45:  54000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:49:  55000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:54:  56000000 \n","INFO  @ Mon, 10 Mar 2025 03:00:58:  57000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:01:  58000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:06:  59000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:10:  60000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:14:  61000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:17:  62000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:22:  63000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:25:  64000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:29:  65000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:33:  66000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:37:  67000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:40:  68000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:44:  69000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:47:  70000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:51:  71000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:54:  72000000 \n","INFO  @ Mon, 10 Mar 2025 03:01:57:  73000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:01:  74000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:04:  75000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:08:  76000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:11:  77000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:15:  78000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:18:  79000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:21:  80000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:25:  81000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:28:  82000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:32:  83000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:35:  84000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:38:  85000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:42:  86000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:45:  87000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:49:  88000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:52:  89000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:55:  90000000 \n","INFO  @ Mon, 10 Mar 2025 03:02:59:  91000000 \n","INFO  @ Mon, 10 Mar 2025 03:03:02:  92000000 \n","INFO  @ Mon, 10 Mar 2025 03:03:13:  93000000 \n","INFO  @ Mon, 10 Mar 2025 03:03:26:  94000000 \n","INFO  @ Mon, 10 Mar 2025 03:03:32: 94172312 reads have been read. \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #1 tag size is determined as 49 bps \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #1 tag size = 49.0 \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #1  total tags in treatment: 94172312 \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #1 finished! \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #2 Build Peak Model... \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #2 Skipped... \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #2 Use 200 as fragment length \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #2 Sequencing ends will be shifted towards 5' by 100 bp(s) \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #3 Call peaks... \n","INFO  @ Mon, 10 Mar 2025 03:03:33: #3 Pre-compute pvalue-qvalue table... \n","INFO  @ Mon, 10 Mar 2025 03:09:53: #3 In the peak calling step, the following will be performed simultaneously: \n","INFO  @ Mon, 10 Mar 2025 03:09:53: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... SRR891268_ATAC_Peaks_treat_pileup.bdg \n","INFO  @ Mon, 10 Mar 2025 03:09:53: #3   Write bedGraph files for control lambda (after scaling if necessary)... SRR891268_ATAC_Peaks_control_lambda.bdg \n","INFO  @ Mon, 10 Mar 2025 03:09:53: #3   --SPMR is requested, so pileup will be normalized by sequencing depth in million reads. \n","INFO  @ Mon, 10 Mar 2025 03:09:53: #3 Call peaks for each chromosome... \n","INFO  @ Mon, 10 Mar 2025 03:11:11: #4 Write output xls file... /content/ATACseq_Peaks/SRR891268_ATAC_Peaks_peaks.xls \n","INFO  @ Mon, 10 Mar 2025 03:11:11: #4 Write peak in narrowPeak format file... /content/ATACseq_Peaks/SRR891268_ATAC_Peaks_peaks.narrowPeak \n","INFO  @ Mon, 10 Mar 2025 03:11:11: #4 Write summits bed file... /content/ATACseq_Peaks/SRR891268_ATAC_Peaks_summits.bed \n","INFO  @ Mon, 10 Mar 2025 03:11:11: Done! \n"]}]},{"cell_type":"code","source":["# List output files\n","!ls -lh /content/ATACseq_Peaks"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"H0Zrfi---9AG","executionInfo":{"status":"ok","timestamp":1741576271921,"user_tz":240,"elapsed":104,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"0cc287b2-6294-4c46-9e69-073ba20a777c"},"execution_count":28,"outputs":[{"output_type":"stream","name":"stdout","text":["total 1.6G\n","-rw-r--r-- 1 root root  92M Mar 10 03:11 SRR891268_ATAC_Peaks_control_lambda.bdg\n","-rw-r--r-- 1 root root 3.1M Mar 10 03:11 SRR891268_ATAC_Peaks_peaks.narrowPeak\n","-rw-r--r-- 1 root root 3.3M Mar 10 03:11 SRR891268_ATAC_Peaks_peaks.xls\n","-rw-r--r-- 1 root root 2.2M Mar 10 03:11 SRR891268_ATAC_Peaks_summits.bed\n","-rw-r--r-- 1 root root 1.5G Mar 10 03:11 SRR891268_ATAC_Peaks_treat_pileup.bdg\n"]}]},{"cell_type":"code","source":["# Convert narrowPeak to BED\n","!cut -f 1-3 /content/ATACseq_Peaks/SRR891268_ATAC_Peaks_peaks.narrowPeak > /content/ATACseq_Peaks/SRR891268_ATAC_Peaks.bed"],"metadata":{"id":"zSYvwb3m_vQ6","executionInfo":{"status":"ok","timestamp":1741576272129,"user_tz":240,"elapsed":206,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}}},"execution_count":29,"outputs":[]},{"cell_type":"code","source":["# Convert bedGraph to BigWig (for genome browsers)\n","!bedGraphToBigWig /content/ATACseq_Peaks/SRR891268_ATAC_Peaks_treat_pileup.bdg \\\n","/content/reference_genome/GRCh38.chrom.sizes \\\n","/content/ATACseq_Peaks/SRR891268_ATAC_Peaks.bw\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"D0C7AtA-_xoF","executionInfo":{"status":"ok","timestamp":1741576272345,"user_tz":240,"elapsed":212,"user":{"displayName":"Veysel Sapan","userId":"06116431774657820768"}},"outputId":"ab4b084a-c2e1-48c2-d502-2354d10021ea"},"execution_count":30,"outputs":[{"output_type":"stream","name":"stdout","text":["/bin/bash: line 1: bedGraphToBigWig: command not found\n"]}]},{"cell_type":"code","source":["# Zip the peak files\n","!zip -r /content/ATACseq_Peaks.zip /content/ATACseq_Peaks\n","\n","# Provide a download link\n","from google.colab import files\n","files.download(\"/content/ATACseq_Peaks.zip\")\n"],"metadata":{"id":"l3lWepjuAA4p"},"execution_count":null,"outputs":[]}],"metadata":{"colab":{"machine_shape":"hm","provenance":[],"mount_file_id":"1MOcuHdxt_b-DdMq5lXMjM4TDbWOWDJNT","authorship_tag":"ABX9TyMh+VLvFD0RW4xFLJGsOwWH"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}