{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "b74c9f57-9664-4f70-96e5-374ddf4ba14f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mAssuming unrestricted shared filesystem usage.\u001b[0m\n",
      "\u001b[32mhost: veysel\u001b[0m\n",
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
      "\u001b[33mProvided cores: 4\u001b[0m\n",
      "\u001b[33mRules claiming more threads will be scaled down.\u001b[0m\n",
      "\u001b[33mJob stats:\n",
      "job                 count\n",
      "----------------  -------\n",
      "all                     1\n",
      "plink_download          1\n",
      "plink_pca               1\n",
      "plot_pca                1\n",
      "subsample_1p            1\n",
      "uncompress_plink        2\n",
      "total                   7\n",
      "\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:26:13 2025]\u001b[0m\n",
      "\u001b[32mlocalrule plink_download:\n",
      "    output: scratch/hapmap.map.gz, scratch/hapmap.ped.gz, data/relationships.txt\n",
      "    jobid: 5\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 10.7M  100 10.7M    0     0  14.4M      0 --:--:-- --:--:-- --:--:-- 14.4M\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 1216M  100 1216M    0     0  35.3M      0  0:00:34  0:00:34 --:--:-- 27.2M\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 39243  100 39243    0     0   126k      0 --:--:-- --:--:-- --:--:--  126k\n",
      "\u001b[32m[Sat Feb  8 05:26:49 2025]\u001b[0m\n",
      "\u001b[32mFinished job 5.\u001b[0m\n",
      "\u001b[32m1 of 7 steps (14%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 2 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:26:49 2025]\u001b[0m\n",
      "\u001b[32mlocalrule uncompress_plink:\n",
      "    input: scratch/hapmap.map.gz\n",
      "    output: data/hapmap.map\n",
      "    jobid: 6\n",
      "    reason: Forced execution\n",
      "    wildcards: plinkext=map\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:26:49 2025]\u001b[0m\n",
      "\u001b[32mlocalrule uncompress_plink:\n",
      "    input: scratch/hapmap.ped.gz\n",
      "    output: data/hapmap.ped\n",
      "    jobid: 4\n",
      "    reason: Forced execution\n",
      "    wildcards: plinkext=ped\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:26:49 2025]\u001b[0m\n",
      "\u001b[32mFinished job 6.\u001b[0m\n",
      "\u001b[32m2 of 7 steps (29%) done\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:27:26 2025]\u001b[0m\n",
      "\u001b[32mFinished job 4.\u001b[0m\n",
      "\u001b[32m3 of 7 steps (43%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:27:26 2025]\u001b[0m\n",
      "\u001b[32mlocalrule subsample_1p:\n",
      "    input: data/hapmap.ped, data/hapmap.map\n",
      "    output: data/hapmap1.ped, data/hapmap1.map\n",
      "    jobid: 3\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "host: veysel\n",
      "PLINK v2.00a6 AVX2 (23 Nov 2023)               www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to data/hapmap1.log.\n",
      "Options in effect:\n",
      "  --export ped\n",
      "  --geno 0.1\n",
      "  --out data/hapmap1\n",
      "  --pedmap data/hapmap\n",
      "  --thin 0.01\n",
      "\n",
      "Start time: Sat Feb  8 05:27:26 2025\n",
      "15773 MiB RAM detected, ~8957 available; reserving 7886 MiB for main workspace.\n",
      "Using up to 8 compute threads.\n",
      "--pedmap: 1457897 variants in .map file.\n",
      "--pedmap: 1397 samples present, genotypes extracted to\n",
      "data/hapmap1-temporary.bed.smaj .\n",
      "Transposing sample-major .bed to data/hapmap1-temporary.pgen , and setting\n",
      "major alleles to provisional-REF.\n",
      "Pass 1/1: transposing and compdone.\n",
      "Transpose complete.\n",
      "--pedmap: data/hapmap1-temporary.pgen + data/hapmap1-temporary.pvar +\n",
      "data/hapmap1-temporary.psam written. .bed.smaj and .fam.tmp temporary files\n",
      "deleted.\n",
      "1397 samples (714 females, 683 males; 1198 founders) loaded from\n",
      "data/hapmap1-temporary.psam.\n",
      "1457897 variants loaded from data/hapmap1-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "--thin: 1443437 variants removed (14460 remaining).\n",
      "Calculating allele frequencies... done.\n",
      "--geno: 0 variants removed due to missing genotype data.\n",
      "14460 variants remaining after main filters.\n",
      "Writing data/hapmap1.map ... done.\n",
      "--export ped pass 1/writing.done.\n",
      "--export ped: data/hapmap1.ped written.\n",
      "End time: Sat Feb  8 05:28:13 2025\n",
      "\u001b[32m[Sat Feb  8 05:28:14 2025]\u001b[0m\n",
      "\u001b[32mFinished job 3.\u001b[0m\n",
      "\u001b[32m4 of 7 steps (57%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:28:14 2025]\u001b[0m\n",
      "\u001b[32mlocalrule plink_pca:\n",
      "    input: data/hapmap1.ped, data/hapmap1.map\n",
      "    output: data/hapmap1.eigenvec, data/hapmap1.eigenval\n",
      "    jobid: 2\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "PLINK v2.00a6 AVX2 (23 Nov 2023)               www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to data/hapmap1.log.\n",
      "Options in effect:\n",
      "  --out data/hapmap1\n",
      "  --pca\n",
      "  --pedmap data/hapmap1\n",
      "\n",
      "Start time: Sat Feb  8 05:28:14 2025\n",
      "15773 MiB RAM detected, ~8980 available; reserving 7886 MiB for main workspace.\n",
      "Using up to 8 compute threads.\n",
      "--pedmap: 14460 variants in .map file.\n",
      "--pedmap: 1397 samples present, genotypes extracted to\n",
      "data/hapmap1-temporary.bed.smaj .\n",
      "Transposing sample-major .bed to data/hapmap1-temporary.pgen , and setting\n",
      "major alleles to provisional-REF.\n",
      "Pass 1/1: transposing and compressing... done.\n",
      "Transpose complete.\n",
      "--pedmap: data/hapmap1-temporary.pgen + data/hapmap1-temporary.pvar +\n",
      "data/hapmap1-temporary.psam written. .bed.smaj and .fam.tmp temporary files\n",
      "deleted.\n",
      "1397 samples (714 females, 683 males; 1198 founders) loaded from\n",
      "data/hapmap1-temporary.psam.\n",
      "14460 variants loaded from data/hapmap1-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "Excluding 297 variants on non-autosomes from GRM construction.\n",
      "done.\n",
      "Correcting for missingness... done.\n",
      "Excluding 297 variants on non-autosomes from PCA.\n",
      "done.cting eigenvalues and eigenvectors... \n",
      "--pca: Eigenvectors written to data/hapmap1.eigenvec , and eigenvalues written\n",
      "to data/hapmap1.eigenval .\n",
      "End time: Sat Feb  8 05:28:19 2025\n",
      "\u001b[32m[Sat Feb  8 05:28:19 2025]\u001b[0m\n",
      "\u001b[32mFinished job 2.\u001b[0m\n",
      "\u001b[32m5 of 7 steps (71%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:28:19 2025]\u001b[0m\n",
      "\u001b[32mlocalrule plot_pca:\n",
      "    input: data/hapmap1.eigenvec, data/hapmap1.eigenval\n",
      "    output: pca.png\n",
      "    jobid: 1\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "/home/veysel/Veysel_Sapan/Bioinformatics_with_Python_Cookbook/Bioinformatics-with-Python-Cookbook-third-edition/Chapter09/snakemake/.snakemake/scripts/tmpc9gb4d6c.plot_pca.py:7: SyntaxWarning: invalid escape sequence '\\s'\n",
      "  pca_df = pd.read_csv(eigen_fname, sep='\\s+')\n",
      "/home/veysel/.local/lib/python3.12/site-packages/matplotlib/projections/__init__.py:63: UserWarning: Unable to import Axes3D. This may be due to multiple versions of Matplotlib being installed (e.g. as a system package and as a pip package). As a result, the 3D projection is not available.\n",
      "  warnings.warn(\"Unable to import Axes3D. This may be due to multiple versions of \"\n",
      "\u001b[32m[Sat Feb  8 05:28:21 2025]\u001b[0m\n",
      "\u001b[32mFinished job 1.\u001b[0m\n",
      "\u001b[32m6 of 7 steps (86%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:28:21 2025]\u001b[0m\n",
      "\u001b[32mlocalrule all:\n",
      "    input: pca.png\n",
      "    jobid: 0\n",
      "    reason: Forced execution\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Sat Feb  8 05:28:21 2025]\u001b[0m\n",
      "\u001b[32mFinished job 0.\u001b[0m\n",
      "\u001b[32m7 of 7 steps (100%) done\u001b[0m\n",
      "\u001b[33mComplete log: .snakemake/log/2025-02-08T052613.518822.snakemake.log\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!snakemake --forceall --cores 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "86811d22-4809-4474-9aa9-8af53857e379",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mAssuming unrestricted shared filesystem usage.\u001b[0m\n",
      "\u001b[32mhost: veysel\u001b[0m\n",
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "\u001b[33mNothing to be done (all requested files are present and up to date).\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!snakemake -c1 data/relationships.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "ebb4a238-243d-4f85-92b7-10c0099dffd5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "/bin/bash: line 1: svg: command not found\n"
     ]
    }
   ],
   "source": [
    "!snakemake --dag data/hapmap1.eigenvec | dot -Tsvg > bio.\n",
    "!svg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20f280ff-0911-43d3-85db-5266853a94d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
